// @arcgis/map-components/dist/cjs/arcgis-navigation-toggle.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const navigationToggleCss = "arcgis-navigation-toggle{}";
const ArcgisNavigationToggleStyle0 = navigationToggleCss;

const useNavigationToggleWidget = useWidget.makeWidgetController(index$2.newWidgetsNavigationToggle);
const ArcgisNavigationToggle = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useNavigationToggleWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("layout", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.layout = this.widget.layout;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Watchers
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async toggle() {
        this.widget?.toggle();
    }
    get el() { return index.getElement(this); }
};
ArcgisNavigationToggle.style = ArcgisNavigationToggleStyle0;

exports.arcgis_navigation_toggle = ArcgisNavigationToggle;
