// @arcgis/map-components/dist/cjs/arcgis-attachments.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const useWidget = require('./useWidget-4ffee640.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
require('./component-utils-387e34d8.js');

const attachmentsCss = "";
const ArcgisAttachmentsStyle0 = attachmentsCss;

const useAttachmentsWidget = useWidget.makeWidgetController(index$2.newWidgetsAttachments);
const ArcgisAttachments = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useAttachmentsWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("submitting", "state");
        this.referenceElement = undefined;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.capabilities = this.widget.capabilities;
        this.displayType = this.widget.displayType;
        this.graphic = this.widget.graphic;
        this.submitting = this.widget.submitting;
        this.hideAddButton = this.widget.visibleElements.addButton;
        this.hideAddSubmitButton = this.widget.visibleElements.addSubmitButton;
        this.hideCancelAddButton = this.widget.visibleElements.cancelAddButton;
        this.hideCancelUpdateButton = this.widget.visibleElements.cancelUpdateButton;
        this.hideDeleteButton = this.widget.visibleElements.deleteButton;
        this.hideErrorMessage = this.widget.visibleElements.errorMessage;
        this.hideProgressBar = this.widget.visibleElements.progressBar;
        this.hideUpdateButton = this.widget.visibleElements.updateButton;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisAttachments.style = ArcgisAttachmentsStyle0;

exports.arcgis_attachments = ArcgisAttachments;
