// @arcgis/map-components/dist/cjs/arcgis-track.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const trackCss = "";
const ArcgisTrackStyle0 = trackCss;

const useTrackWidget = useWidget.makeWidgetController(index$2.newWidgetsTrack);
const ArcgisTrack = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.widgetTrack = index.createEvent(this, "arcgisComplete", 7);
        this.widgetTrackError = index.createEvent(this, "arcgisError", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useTrackWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.widget.geolocationOptions;
        this.goToLocationEnabled = this.widget.goToLocationEnabled;
        this.graphic = this.widget.graphic;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.positionFilterFunction = this.viewModel.positionFilterFunction;
        this.rotationEnabled = this.widget.rotationEnabled;
        this.scale = this.widget.scale;
        this.tracking = this.widget.tracking;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget.start();
    }
    async stop() {
        this.widget.stop();
    }
    get el() { return index.getElement(this); }
};
ArcgisTrack.style = ArcgisTrackStyle0;

exports.arcgis_track = ArcgisTrack;
