// @arcgis/map-components/dist/cjs/arcgis-slider.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const sliderCss = "arcgis-slider{min-width:200px;min-height:25px}";
const ArcgisSliderStyle0 = sliderCss;

const useSliderWidget = useWidget.makeWidgetController(index$2.newWidgetsSlider);
const ArcgisSlider = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.widgetMaxChange = index.createEvent(this, "arcgisMaxChange", 7);
        this.widgetMaxClick = index.createEvent(this, "arcgisMaxClick", 7);
        this.widgetMinChange = index.createEvent(this, "arcgisMinChange", 7);
        this.widgetMinClick = index.createEvent(this, "arcgisMinClick", 7);
        this.widgetSegmentClick = index.createEvent(this, "arcgisSegmentClick", 7);
        this.widgetSegmentDrag = index.createEvent(this, "arcgisSegmentDrag", 7);
        this.widgetThumbChange = index.createEvent(this, "arcgisThumbChange", 7);
        this.widgetThumbClick = index.createEvent(this, "arcgisThumbClick", 7);
        this.widgetThumbDrag = index.createEvent(this, "arcgisThumbDrag", 7);
        this.widgetTickClick = index.createEvent(this, "arcgisTickClick", 7);
        this.widgetTrackClick = index.createEvent(this, "arcgisTrackClick", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useSliderWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.draggableSegmentsEnabled = this.widget.draggableSegmentsEnabled;
        this.effectiveMax = this.widget.effectiveMax;
        this.effectiveMin = this.widget.effectiveMin;
        this.effectiveSegmentElements = this.widget.effectiveSegmentElements;
        this.labelElements = this.widget.labelElements;
        this.labelInputsEnabled = this.widget.labelInputsEnabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.labels = this.widget.labels;
        this.layout = this.widget.layout;
        this.min = this.widget.min;
        this.max = this.widget.max;
        this.maxLabelElement = this.widget.maxLabelElement;
        this.minLabelElement = this.widget.minLabelElement;
        this.precision = this.widget.precision;
        this.rangeLabelInputsEnabled = this.widget.rangeLabelInputsEnabled;
        this.segmentElements = this.widget.segmentElements;
        this.snapOnClickEnabled = this.widget.snapOnClickEnabled;
        this.syncedSegmentsEnabled = this.widget.syncedSegmentsEnabled;
        this.thumbElements = this.widget.thumbElements;
        this.thumbsConstrained = this.widget.thumbsConstrained;
        this.tickConfigs = this.widget.tickConfigs;
        this.tickElements = this.widget.tickElements;
        this.trackElement = this.widget.trackElement;
        this.steps = this.manager.getSet(this.widget.steps, {
            set: (newValue) => (typeof newValue === "string" ? newValue.split(",").map(Number) : newValue),
        });
        this.values = this.manager.getSet(this.widget.values, {
            set: (newValue) => (typeof newValue === "string" ? newValue.split(",").map(Number) : newValue),
        });
        this.position = "bottom-left";
        this.state = this.widget.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisSlider.style = ArcgisSliderStyle0;

exports.arcgis_slider = ArcgisSlider;
