// @arcgis/map-components/dist/cjs/arcgis-value-picker.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const useWidget = require('./useWidget-4ffee640.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
require('./component-utils-387e34d8.js');

const valuePickerCss = "arcgis-value-picker{}";
const ArcgisValuePickerStyle0 = valuePickerCss;

const useValuePickerWidget = useWidget.makeWidgetController(index$2.newWidgetsValuePicker);
const ArcgisValuePicker = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.widgetAnimate = index.createEvent(this, "arcgisAnimate", 7);
        this.widgetNext = index.createEvent(this, "arcgisNext", 7);
        this.widgetPause = index.createEvent(this, "arcgisPause", 7);
        this.widgetPlay = index.createEvent(this, "arcgisPlay", 7);
        this.widgetPrevious = index.createEvent(this, "arcgisPrevious", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useValuePickerWidget(this);
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("values");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.position = "bottom-left";
        this.canNext = this.widget.canNext;
        this.canPlay = this.widget.canPlay;
        this.canPrevious = this.widget.canPrevious;
        this.caption = this.widget.caption;
        this.component = this.widget.component;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layout = this.widget.layout;
        this.loop = this.widget.loop;
        this.playRate = this.widget.playRate;
        this.hideNextButton = this.widget.visibleElements.nextButton;
        this.hidePlayButton = this.widget.visibleElements.playButton;
        this.hidePreviousButton = this.widget.visibleElements.previousButton;
        this.values = this.widget.values;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async next() {
        this.widget?.next();
    }
    async pause() {
        this.widget?.pause();
    }
    async play() {
        this.widget?.play();
    }
    async previous() {
        this.widget?.previous();
    }
    get el() { return index.getElement(this); }
};
ArcgisValuePicker.style = ArcgisValuePickerStyle0;

exports.arcgis_value_picker = ArcgisValuePicker;
