// @arcgis/map-components/dist/cjs/arcgis-layer-list.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const layerListCss = "";
const ArcgisLayerListStyle0 = layerListCss;

const useLayerListWidget = useWidget.makeWidgetController(index$2.newWidgetsLayerList);
const ArcgisLayerList = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.widgetTriggerAction = index.createEvent(this, "arcgisTriggerAction", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useLayerListWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.collapsed = this.widget.collapsed;
        this.headingLevel = this.widget.headingLevel;
        this.catalogLayerList = this.widget.catalogLayerList;
        this.catalogOptions = this.widget.catalogOptions;
        this.dragEnabled = this.widget.dragEnabled;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.knowledgeGraphOptions = this.widget.knowledgeGraphOptions;
        this.tableList = this.widget.tableList;
        this.label = this.widget.label;
        this.listItemCreatedFunction = this.widget.listItemCreatedFunction;
        this.minDragEnabledItems = this.widget.minDragEnabledItems;
        this.minFilterItems = this.widget.minFilterItems;
        this.selectedItems = this.widget.selectedItems;
        this.selectionMode = this.widget.selectionMode;
        this.visibilityAppearance = this.widget.visibilityAppearance;
        this.hideCatalogLayerList = this.widget.visibleElements.catalogLayerList;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showErrors = this.widget.visibleElements.errors;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.showHeading = this.widget.visibleElements.heading;
        this.hideStatusIndicators = this.widget.visibleElements.statusIndicators;
        this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async triggerAction(action, item) {
        this.widget?.triggerAction(action, item);
    }
    get el() { return index.getElement(this); }
};
ArcgisLayerList.style = ArcgisLayerListStyle0;

exports.arcgis_layer_list = ArcgisLayerList;
