// @arcgis/map-components/dist/cjs/arcgis-building-explorer.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const useWidget = require('./useWidget-4ffee640.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
require('./component-utils-387e34d8.js');

const buildingExplorerCss = "";
const ArcgisBuildingExplorerStyle0 = buildingExplorerCss;

const useBuildingExplorerWidget = useWidget.makeWidgetController(index$2.newWidgetsBuildingExplorer);
const ArcgisBuildingExplorer = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useBuildingExplorerWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("level", "phase", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layers = this.widget.layers;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.viewModel.level;
        this.phase = this.viewModel.phase;
        this.state = this.viewModel.state;
        this.hideLevels = this.widget.visibleElements.levels;
        this.hidePhases = this.widget.visibleElements.phases;
        this.hideDisciplines = this.widget.visibleElements.disciplines;
        this.position = "bottom-left";
        this.referenceElement = undefined;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisBuildingExplorer.style = ArcgisBuildingExplorerStyle0;

exports.arcgis_building_explorer = ArcgisBuildingExplorer;
