// @arcgis/map-components/dist/cjs/arcgis-time-slider.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const timeSliderCss = "arcgis-time-slider{}";
const ArcgisTimeSliderStyle0 = timeSliderCss;

const useTimeSliderWidget = useWidget.makeWidgetController(index$2.newWidgetsTimeSlider);
const ArcgisTimeSlider = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.widgetTriggerAction = index.createEvent(this, "arcgisTriggerAction", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useTimeSliderWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("effectiveStops", "fullTimeExtent", "timeExtent", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.actions = this.widget.actions;
        this.disabled = this.widget.disabled;
        this.effectiveStops = this.widget.effectiveStops;
        this.fullTimeExtent = this.widget.fullTimeExtent;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layout = this.widget.layout;
        this.loop = this.widget.loop;
        this.mode = this.widget.mode;
        this.playRate = this.widget.playRate;
        this.stops = this.widget.stops;
        this.tickConfigs = this.widget.tickConfigs;
        this.timeExtent = this.widget.timeExtent;
        this.timeVisible = this.widget.timeVisible;
        this.timeZone = this.widget.timeZone;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async applyTimeSliderSettings(settings) {
        Object.assign(this, settings);
    }
    async next() {
        this.widget?.next();
    }
    async play() {
        this.widget?.play();
    }
    async previous() {
        this.widget?.previous();
    }
    async stop() {
        this.widget?.stop();
    }
    async updateWebDocument(webmap) {
        this.widget?.updateWebDocument(webmap);
    }
    get el() { return index.getElement(this); }
};
ArcgisTimeSlider.style = ArcgisTimeSliderStyle0;

exports.arcgis_time_slider = ArcgisTimeSlider;
