// @arcgis/map-components/dist/cjs/arcgis-placement.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const componentUtils = require('./component-utils-387e34d8.js');

const placementCss = "arcgis-placement{display:block;pointer-events:auto;box-shadow:none}.arcgis-placement__container{display:flex}";
const ArcgisPlacementStyle0 = placementCss;

const ArcgisPlacement = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this._setContainer = (element) => {
            this.el.childElem = element;
        };
        this.referenceElement = undefined;
        this.position = "bottom-left";
    }
    referenceElementWatcher(newValue) {
        void componentUtils.referenceElementHelper(this, newValue);
    }
    positionWatcher(newValue) {
        if (this.el.childElem && newValue !== undefined) {
            this.el.view?.ui.move(this.el.childElem, newValue);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    async componentDidLoad() {
        componentUtils.defineViewProperty(this);
        componentUtils.attachToParent(this);
    }
    disconnectedCallback() {
        if (this.el.childElem) {
            this.el.view?.ui?.remove(this.el.childElem);
        }
    }
    render() {
        return (index.h("div", { key: '6a5216cd50b19569d307a87af6e1d1c1e938d096', ref: this._setContainer, class: "arcgis-placement__container" }, index.h("slot", { key: 'e7f78de5ce1621f0aafa7f84005b60ca66cbbc8c' })));
    }
    get el() { return index.getElement(this); }
    static get watchers() { return {
        "referenceElement": ["referenceElementWatcher"],
        "position": ["positionWatcher"]
    }; }
};
ArcgisPlacement.style = ArcgisPlacementStyle0;

exports.arcgis_placement = ArcgisPlacement;
