// @arcgis/map-components/dist/cjs/arcgis-locate.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const locateCss = "arcgis-locate{}";
const ArcgisLocateStyle0 = locateCss;

const useLocateWidget = useWidget.makeWidgetController(index$2.newWidgetsLocate);
const ArcgisLocate = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.widgetLocate = index.createEvent(this, "arcgisSuccess", 7);
        this.widgetLocateError = index.createEvent(this, "arcgisError", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useLocateWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.widget.geolocationOptions;
        this.goToLocationDisabled = this.widget.goToLocationEnabled;
        this.graphic = this.widget.graphic;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.popupDisabled = this.widget.popupEnabled;
        this.scale = this.widget.scale;
        this.state = this.viewModel.state;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async cancelLocate() {
        this.widget?.cancelLocate();
    }
    async locate() {
        return await this.widget?.locate();
    }
    get el() { return index.getElement(this); }
};
ArcgisLocate.style = ArcgisLocateStyle0;

exports.arcgis_locate = ArcgisLocate;
