// @arcgis/map-components/dist/cjs/arcgis-scale-bar.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const scaleBarCss = "arcgis-scale-bar{}";
const ArcgisScaleBarStyle0 = scaleBarCss;

const useScaleBarWidget = useWidget.makeWidgetController(index$2.newWidgetsScaleBar);
const ArcgisScaleBar = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useScaleBarWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.barStyle = this.widget.style;
        this.unit = this.widget.unit;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisScaleBar.style = ArcgisScaleBarStyle0;

exports.arcgis_scale_bar = ArcgisScaleBar;
