// @arcgis/map-components/dist/cjs/arcgis-zoom.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const index$2 = require('./index-c3c4ddd9.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const zoomCss = "arcgis-zoom{}";
const ArcgisZoomStyle0 = zoomCss;

const useZoomWidget = useWidget.makeWidgetController(index$2.newWidgetsZoom);
const ArcgisZoom = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        //--------------------------------------------------------------------------
        //
        //  Properties
        //
        //--------------------------------------------------------------------------
        this.widget = useZoomWidget(this);
        this.viewModel = this.widget.viewModel;
        this.reactiveUtils = chunkPVEVZB4O.load(index$2.importCoreReactiveUtils);
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layout = this.widget.layout;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
        this.autoDestroyDisabled = false;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async zoomIn() {
        this.widget?.viewModel.zoomIn();
    }
    async zoomOut() {
        this.widget?.viewModel.zoomOut();
    }
    get el() { return index.getElement(this); }
};
ArcgisZoom.style = ArcgisZoomStyle0;

exports.arcgis_zoom = ArcgisZoom;
