// @arcgis/map-components/dist/cjs/arcgis-bookmarks.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-eee1d83b.js');
const index$2 = require('./index-c3c4ddd9.js');
const chunkPVEVZB4O = require('./chunk-PVEVZB4O-270e4213.js');
const index$1 = require('./index-8d100f9f.js');
const useWidget = require('./useWidget-4ffee640.js');
require('./component-utils-387e34d8.js');

const bookmarksCss = "";
const ArcgisBookmarksStyle0 = bookmarksCss;

const useBookmarksWidget = useWidget.makeWidgetController(index$2.newWidgetsBookmarks);
const ArcgisBookmarks = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.widgetBookmarkEdit = index.createEvent(this, "arcgisEdit", 7);
        this.widgetBookmarkSelect = index.createEvent(this, "arcgisSelect", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useBookmarksWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = chunkPVEVZB4O.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.bookmarks = this.widget.bookmarks;
        this.defaultCreateOptions = this.widget.defaultCreateOptions;
        this.defaultEditOptions = this.widget.defaultEditOptions;
        this.disabled = this.widget.disabled;
        this.dragEnabled = this.widget.dragEnabled;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.showAddBookmarkButton = this.widget.visibleElements.addBookmarkButton;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showEditBookmarkButton = this.widget.visibleElements.editBookmarkButton;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.showHeading = this.widget.visibleElements.heading;
        this.hideThumbnail = this.widget.visibleElements.thumbnail;
        this.hideTime = this.widget.visibleElements.time;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisBookmarks.style = ArcgisBookmarksStyle0;

exports.arcgis_bookmarks = ArcgisBookmarks;
