// @arcgis/map-components/dist/esm/arcgis-directions.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { z as newLayersRouteLayer, A as newWidgetsDirections } from './index-d2df902e.js';
import { b as isNullOrEmptyString } from './component-utils-de9339fe.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';

const directionsCss = "";
const ArcgisDirectionsStyle0 = directionsCss;

const useDirectionsWidget = makeWidgetController(newWidgetsDirections);
const loggerName = "map-components:arcgis-directions";
const ArcgisDirections = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useDirectionsWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("lastRoute", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.apiKey = this.widget.apiKey;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.routeLayerItemId = this.manager.getSet(undefined, {
            set: (value) => {
                if (isNullOrEmptyString(value)) {
                    this.widget.view?.map.remove(this.layer);
                    this._setDefaultLayer();
                }
                else if (value && this.widget) {
                    newLayersRouteLayer({
                        portalItem: {
                            id: value,
                        },
                    })
                        .then((layer) => {
                        this.layer = layer;
                    })
                        .catch(console.error);
                }
                return value;
            },
        });
        this.lastRoute = this.widget.lastRoute;
        this.layer = this.widget.layer;
        this.headingLevel = this.widget.headingLevel;
        this.maxStops = this.widget.maxStops;
        this.searchProperties = this.widget.searchProperties;
        this.unit = this.widget.unit;
        this.useDefaultRouteLayer = this.manager.getSet(false, {
            set: (newValue) => {
                this._setDefaultLayer();
                return newValue;
            },
        });
        this.hideLayerDetails = this.widget.visibleElements.layerDetails;
        this.hideSaveAsButton = this.widget.visibleElements.saveAsButton;
        this.hideSaveButton = this.widget.visibleElements.saveButton;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Private Methods
    //
    //--------------------------------------------------------------------------
    /** Maintain a default route layer */
    _setDefaultLayer() {
        if (this.useDefaultRouteLayer) {
            newLayersRouteLayer({})
                .then((layer) => {
                this.layer = layer;
            })
                .catch(console.error);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async getDirections() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to get directions`);
        }
        return await this.widget?.getDirections();
    }
    async save() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions`);
        }
        return await this.widget?.save();
    }
    async saveAs(portalItem, options) {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions as a Portal Item`);
        }
        return await this.widget?.saveAs(portalItem, options);
    }
    async zoomToRoute() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to zoom to route`);
        }
        this.widget?.zoomToRoute();
    }
    get el() { return getElement(this); }
};
ArcgisDirections.style = ArcgisDirectionsStyle0;

export { ArcgisDirections as arcgis_directions };
