// @arcgis/map-components/dist/esm/arcgis-line-of-sight.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { U as newWidgetsLineOfSight } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const lineOfSightCss = "arcgis-line-of-sight{}";
const ArcgisLineOfSightStyle0 = lineOfSightCss;

const useLineOfSightWidget = makeWidgetController(newWidgetsLineOfSight);
const ArcgisLineOfSight = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useLineOfSightWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async clear() {
        this.widget?.viewModel.clear();
    }
    async continue() {
        this.widget?.viewModel.continue();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    async stop() {
        this.widget?.viewModel.stop();
    }
    get el() { return getElement(this); }
};
ArcgisLineOfSight.style = ArcgisLineOfSightStyle0;

export { ArcgisLineOfSight as arcgis_line_of_sight };
