// @arcgis/map-components/dist/esm/arcgis-basemap-toggle.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { i as importCoreReactiveUtils, r as newWidgetsBasemapToggle } from './index-d2df902e.js';
import { l as load, u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const basemapToggleCss = "";
const ArcgisBasemapToggleStyle0 = basemapToggleCss;

const useBasemapToggleWidget = makeWidgetController(newWidgetsBasemapToggle);
const ArcgisBasemapToggle = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.reactiveUtils = load(importCoreReactiveUtils);
        this.widget = useBasemapToggleWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("nextBasemap", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.referenceElement = undefined;
        this.nextBasemap = this.widget.nextBasemap;
        this.showTitle = this.widget.visibleElements.title;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async toggle() {
        return await this.widget?.toggle();
    }
    get el() { return getElement(this); }
};
ArcgisBasemapToggle.style = ArcgisBasemapToggleStyle0;

export { ArcgisBasemapToggle as arcgis_basemap_toggle };
