// @arcgis/map-components/dist/esm/arcgis-track.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { ab as newWidgetsTrack } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const trackCss = "";
const ArcgisTrackStyle0 = trackCss;

const useTrackWidget = makeWidgetController(newWidgetsTrack);
const ArcgisTrack = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.widgetTrack = createEvent(this, "arcgisComplete", 7);
        this.widgetTrackError = createEvent(this, "arcgisError", 7);
        this.manager = useControllerManager(this);
        this.widget = useTrackWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.widget.geolocationOptions;
        this.goToLocationEnabled = this.widget.goToLocationEnabled;
        this.graphic = this.widget.graphic;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.positionFilterFunction = this.viewModel.positionFilterFunction;
        this.rotationEnabled = this.widget.rotationEnabled;
        this.scale = this.widget.scale;
        this.tracking = this.widget.tracking;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget.start();
    }
    async stop() {
        this.widget.stop();
    }
    get el() { return getElement(this); }
};
ArcgisTrack.style = ArcgisTrackStyle0;

export { ArcgisTrack as arcgis_track };
