// @arcgis/map-components/dist/esm/arcgis-legend.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { T as newWidgetsLegend } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const legendCss = "";
const ArcgisLegendStyle0 = legendCss;

const useLegendWidget = makeWidgetController(newWidgetsLegend);
const ArcgisLegend = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useLegendWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.headingLevel = this.widget.headingLevel;
        this.referenceElement = undefined;
        this.activeLayerInfos = this.widget.activeLayerInfos;
        this.basemapLegendVisible = this.widget.basemapLegendVisible;
        this.hideLayersNotInCurrentView = this.widget.hideLayersNotInCurrentView;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.respectLayerVisibilityDisabled = this.widget.respectLayerVisibility;
        this.legendStyle = this.widget.style;
        this.layerInfos = this.widget.layerInfos;
        this.position = "bottom-left";
        this.respectLayerDefinitionExpression = this.widget.respectLayerDefinitionExpression;
        this.state = this.viewModel.state;
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisLegend.style = ArcgisLegendStyle0;

export { ArcgisLegend as arcgis_legend };
