// @arcgis/map-components/dist/esm/arcgis-directional-pad.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { y as newWidgetsDirectionalPad } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const directionalPadCss = "arcgis-directional-pad{}";
const ArcgisDirectionalPadStyle0 = directionalPadCss;

const useDirectionalPadWidget = makeWidgetController(newWidgetsDirectionalPad);
const ArcgisDirectionalPad = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useDirectionalPadWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.disabled = this.widget.disabled;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.visualScale = this.widget.visualScale;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisDirectionalPad.style = ArcgisDirectionalPadStyle0;

export { ArcgisDirectionalPad as arcgis_directional_pad };
