// @arcgis/map-components/dist/esm/arcgis-bookmarks.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { t as newWidgetsBookmarks } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const bookmarksCss = "";
const ArcgisBookmarksStyle0 = bookmarksCss;

const useBookmarksWidget = makeWidgetController(newWidgetsBookmarks);
const ArcgisBookmarks = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetBookmarkEdit = createEvent(this, "arcgisEdit", 7);
        this.widgetBookmarkSelect = createEvent(this, "arcgisSelect", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useBookmarksWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.bookmarks = this.widget.bookmarks;
        this.defaultCreateOptions = this.widget.defaultCreateOptions;
        this.defaultEditOptions = this.widget.defaultEditOptions;
        this.disabled = this.widget.disabled;
        this.dragEnabled = this.widget.dragEnabled;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.showAddBookmarkButton = this.widget.visibleElements.addBookmarkButton;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showEditBookmarkButton = this.widget.visibleElements.editBookmarkButton;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.showHeading = this.widget.visibleElements.heading;
        this.hideThumbnail = this.widget.visibleElements.thumbnail;
        this.hideTime = this.widget.visibleElements.time;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisBookmarks.style = ArcgisBookmarksStyle0;

export { ArcgisBookmarks as arcgis_bookmarks };
