// @arcgis/map-components/dist/esm/arcgis-scale-bar.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { _ as newWidgetsScaleBar } from './index-d2df902e.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const scaleBarCss = "arcgis-scale-bar{}";
const ArcgisScaleBarStyle0 = scaleBarCss;

const useScaleBarWidget = makeWidgetController(newWidgetsScaleBar);
const ArcgisScaleBar = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this);
        this.widget = useScaleBarWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.barStyle = this.widget.style;
        this.unit = this.widget.unit;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisScaleBar.style = ArcgisScaleBarStyle0;

export { ArcgisScaleBar as arcgis_scale_bar };
