// @arcgis/map-components/dist/esm/arcgis-coordinate-conversion.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { w as newWidgetsCoordinateConversion } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const coordinateConversionCss = "";
const ArcgisCoordinateConversionStyle0 = coordinateConversionCss;

const useCoordinateConversionWidget = makeWidgetController(newWidgetsCoordinateConversion);
const ArcgisCoordinateConversion = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useCoordinateConversionWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("currentLocation", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.conversions = this.widget.conversions;
        this.currentLocation = this.widget.currentLocation;
        this.formats = this.widget.formats;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.locationSymbol = this.widget.locationSymbol;
        this.headingLevel = this.widget.headingLevel;
        this.mode = this.widget.mode;
        this.multipleConversionsDisabled = this.widget.multipleConversions;
        this.orientation = this.widget.orientation;
        this.storageDisabled = this.widget.storageEnabled;
        this.storageType = this.widget.storageType;
        this.hideSettingsButton = this.widget.visibleElements.settingsButton;
        this.hideEditButton = this.widget.visibleElements.editButton;
        this.hideExpandButton = this.widget.visibleElements.expandButton;
        this.hideCaptureButton = this.widget.visibleElements.captureButton;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async reverseConvert(coordinate, format) {
        return await this.widget?.reverseConvert(coordinate, format);
    }
    get el() { return getElement(this); }
};
ArcgisCoordinateConversion.style = ArcgisCoordinateConversionStyle0;

export { ArcgisCoordinateConversion as arcgis_coordinate_conversion };
