// @arcgis/map-components/dist/esm/arcgis-daylight.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { x as newWidgetsDaylight } from './index-d2df902e.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const daylightCss = "";
const ArcgisDaylightStyle0 = daylightCss;

const useDaylightWidget = makeWidgetController(newWidgetsDaylight);
const ArcgisDaylight = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this);
        this.widget = useDaylightWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.currentSeason = this.viewModel.currentSeason;
        this.dayPlaying = this.viewModel.dayPlaying;
        this.yearPlaying = this.viewModel.yearPlaying;
        this.dateOrSeason = this.widget.dateOrSeason;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.playSpeedMultiplier = this.widget.playSpeedMultiplier;
        this.timeSliderSteps = this.widget.timeSliderSteps;
        this.hideHeader = this.widget.visibleElements.header;
        this.hidePlayButtons = this.widget.visibleElements.playButtons;
        this.hideDatePicker = this.widget.visibleElements.datePicker;
        this.hideTimezone = this.widget.visibleElements.timezone;
        this.hideSunLightingToggle = this.widget.visibleElements.sunLightingToggle;
        this.hideShadowsToggle = this.widget.visibleElements.shadowsToggle;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisDaylight.style = ArcgisDaylightStyle0;

export { ArcgisDaylight as arcgis_daylight };
