// @arcgis/map-components/dist/esm/arcgis-basemap-gallery.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { p as newWidgetsBasemapGallery } from './index-d2df902e.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import './component-utils-de9339fe.js';

const basemapGalleryCss = "";
const ArcgisBasemapGalleryStyle0 = basemapGalleryCss;

const useBasemapGalleryWidget = makeWidgetController(newWidgetsBasemapGallery);
const ArcgisBasemapGallery = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useBasemapGalleryWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("activeBasemap", "state");
        this.activeBasemap = this.widget.activeBasemap;
        this.autoDestroyDisabled = false;
        this.disabled = this.widget.disabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.source = this.widget.source;
        this.position = "bottom-left";
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisBasemapGallery.style = ArcgisBasemapGalleryStyle0;

export { ArcgisBasemapGallery as arcgis_basemap_gallery };
