// @arcgis/map-components/dist/esm/arcgis-area-measurement-3d.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { m as newWidgetsAreaMeasurement3D } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const areaMeasurement3dCss = "";
const ArcgisAreaMeasurement3dStyle0 = areaMeasurement3dCss;

const useAreaMeasurement3dWidget = makeWidgetController(newWidgetsAreaMeasurement3D);
const ArcgisAreaMeasurement3d = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useAreaMeasurement3dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async clear() {
        this.widget?.viewModel.clear();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    get el() { return getElement(this); }
};
ArcgisAreaMeasurement3d.style = ArcgisAreaMeasurement3dStyle0;

export { ArcgisAreaMeasurement3d as arcgis_area_measurement_3d };
