// @arcgis/map-components/dist/esm/arcgis-placement.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, h, g as getElement } from './index-7d33cc26.js';
import { r as referenceElementHelper, d as defineViewProperty, a as attachToParent } from './component-utils-de9339fe.js';

const placementCss = "arcgis-placement{display:block;pointer-events:auto;box-shadow:none}.arcgis-placement__container{display:flex}";
const ArcgisPlacementStyle0 = placementCss;

const ArcgisPlacement = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this._setContainer = (element) => {
            this.el.childElem = element;
        };
        this.referenceElement = undefined;
        this.position = "bottom-left";
    }
    referenceElementWatcher(newValue) {
        void referenceElementHelper(this, newValue);
    }
    positionWatcher(newValue) {
        if (this.el.childElem && newValue !== undefined) {
            this.el.view?.ui.move(this.el.childElem, newValue);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    async componentDidLoad() {
        defineViewProperty(this);
        attachToParent(this);
    }
    disconnectedCallback() {
        if (this.el.childElem) {
            this.el.view?.ui?.remove(this.el.childElem);
        }
    }
    render() {
        return (h("div", { key: '6a5216cd50b19569d307a87af6e1d1c1e938d096', ref: this._setContainer, class: "arcgis-placement__container" }, h("slot", { key: 'e7f78de5ce1621f0aafa7f84005b60ca66cbbc8c' })));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "referenceElement": ["referenceElementWatcher"],
        "position": ["positionWatcher"]
    }; }
};
ArcgisPlacement.style = ArcgisPlacementStyle0;

export { ArcgisPlacement as arcgis_placement };
