// @arcgis/map-components/dist/esm/arcgis-feature.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { H as newWidgetsFeature } from './index-d2df902e.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import './component-utils-de9339fe.js';

const featureCss = "";
const ArcgisFeatureStyle0 = featureCss;

const useFeatureWidget = makeWidgetController(newWidgetsFeature);
const ArcgisFeature = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useFeatureWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.referenceElement = undefined;
        this.defaultPopupTemplateEnabled = this.widget.defaultPopupTemplateEnabled;
        this.graphic = this.widget.graphic;
        this.location = this.viewModel.location;
        this.map = this.widget.map;
        this.spatialReference = this.widget.spatialReference;
        this.hideTitle = this.widget.visibleElements.title;
        this.hideContentAttachments = this.widget.visibleElements.content.attachments;
        this.hideContentFields = this.widget.visibleElements.content.fields;
        this.hideContentMedia = this.widget.visibleElements.content.media;
        this.hideContentText = this.widget.visibleElements.content.text;
        this.hideContentExpression = this.widget.visibleElements.content.expression;
        this.hideLastEditInfo = this.widget.visibleElements.lastEditedInfo;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async nextMedia(contentElementIndex) {
        this.widget?.nextMedia(contentElementIndex);
    }
    async previousMedia(contentElementIndex) {
        this.widget?.previousMedia(contentElementIndex);
    }
    async setActiveMedia(contentElementIndex, mediaInfoIndex) {
        this.widget?.setActiveMedia(contentElementIndex, mediaInfoIndex);
    }
    get el() { return getElement(this); }
};
ArcgisFeature.style = ArcgisFeatureStyle0;

export { ArcgisFeature as arcgis_feature };
