// @arcgis/map-components/dist/esm/arcgis-swipe.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { a8 as newWidgetsSwipe } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const swipeCss = "arcgis-swipe{}";
const ArcgisSwipeStyle0 = swipeCss;

const useSwipeWidget = makeWidgetController(newWidgetsSwipe);
const ArcgisSwipe = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useSwipeWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.direction = this.widget.direction;
        this.dragLabel = this.widget.dragLabel;
        this.leadingLayers = this.widget.leadingLayers;
        this.swipePosition = this.widget.position;
        this.trailingLayers = this.widget.trailingLayers;
        this.position = "manual";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisSwipe.style = ArcgisSwipeStyle0;

export { ArcgisSwipe as arcgis_swipe };
