// @arcgis/map-components/dist/esm/arcgis-histogram.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { O as newWidgetsHistogram } from './index-d2df902e.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import './component-utils-de9339fe.js';

const histogramCss = "arcgis-histogram{}";
const ArcgisHistogramStyle0 = histogramCss;

const useHistogramWidget = makeWidgetController(newWidgetsHistogram);
const ArcgisHistogram = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useHistogramWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.average = this.widget.average;
        this.barCreatedFunction = this.widget.barCreatedFunction;
        this.bins = this.widget.bins;
        this.dataLineCreatedFunction = this.widget.dataLineCreatedFunction;
        this.dataLines = this.widget.dataLines;
        this.labelFormatFunction = this.widget.labelFormatFunction;
        this.layout = this.widget.layout;
        this.max = this.widget.max;
        this.min = this.widget.min;
        this.position = "bottom-left";
        this.state = this.widget.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisHistogram.style = ArcgisHistogramStyle0;

export { ArcgisHistogram as arcgis_histogram };
