// @arcgis/map-components/dist/esm/arcgis-zoom.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { l as load, u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { i as importCoreReactiveUtils, ai as newWidgetsZoom } from './index-d2df902e.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const zoomCss = "arcgis-zoom{}";
const ArcgisZoomStyle0 = zoomCss;

const useZoomWidget = makeWidgetController(newWidgetsZoom);
const ArcgisZoom = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        //--------------------------------------------------------------------------
        //
        //  Properties
        //
        //--------------------------------------------------------------------------
        this.widget = useZoomWidget(this);
        this.viewModel = this.widget.viewModel;
        this.reactiveUtils = load(importCoreReactiveUtils);
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layout = this.widget.layout;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
        this.autoDestroyDisabled = false;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async zoomIn() {
        this.widget?.viewModel.zoomIn();
    }
    async zoomOut() {
        this.widget?.viewModel.zoomOut();
    }
    get el() { return getElement(this); }
};
ArcgisZoom.style = ArcgisZoomStyle0;

export { ArcgisZoom as arcgis_zoom };
