// @arcgis/map-components/dist/esm/arcgis-locate.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-7d33cc26.js';
import { V as newWidgetsLocate } from './index-d2df902e.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O-2c3b8893.js';
import { u as useControllerManager } from './index-c99fc80c.js';
import { m as makeWidgetController } from './useWidget-9b153437.js';
import './component-utils-de9339fe.js';

const locateCss = "arcgis-locate{}";
const ArcgisLocateStyle0 = locateCss;

const useLocateWidget = makeWidgetController(newWidgetsLocate);
const ArcgisLocate = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetLocate = createEvent(this, "arcgisSuccess", 7);
        this.widgetLocateError = createEvent(this, "arcgisError", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useLocateWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.widget.geolocationOptions;
        this.goToLocationDisabled = this.widget.goToLocationEnabled;
        this.graphic = this.widget.graphic;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.popupDisabled = this.widget.popupEnabled;
        this.scale = this.widget.scale;
        this.state = this.viewModel.state;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async cancelLocate() {
        this.widget?.cancelLocate();
    }
    async locate() {
        return await this.widget?.locate();
    }
    get el() { return getElement(this); }
};
ArcgisLocate.style = ArcgisLocateStyle0;

export { ArcgisLocate as arcgis_locate };
