// @arcgis/map-components/dist/components/arcgis-locate.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { T as newWidgetsLocate } from './index2.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O.js';
import { u as useControllerManager } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';

const locateCss = "arcgis-locate{}";
const ArcgisLocateStyle0 = locateCss;

const useLocateWidget = makeWidgetController(newWidgetsLocate);
const __cmpMeta = [0,"arcgis-locate",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"error":[1032],"geolocationOptions":[1032,"geolocation-options"],"goToLocationDisabled":[1028,"go-to-location-disabled"],"graphic":[1040],"icon":[1025],"label":[1025],"popupDisabled":[1028,"popup-disabled"],"scale":[1026],"state":[1537],"position":[1025],"destroy":[64],"cancelLocate":[64],"locate":[64]}];
const ArcgisLocate$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisLocate extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetLocate = createEvent(this, "arcgisSuccess", 7);
        this.widgetLocateError = createEvent(this, "arcgisError", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useLocateWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.widget.geolocationOptions;
        this.goToLocationDisabled = this.widget.goToLocationEnabled;
        this.graphic = this.widget.graphic;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.popupDisabled = this.widget.popupEnabled;
        this.scale = this.widget.scale;
        this.state = this.viewModel.state;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async cancelLocate() {
        this.widget?.cancelLocate();
    }
    async locate() {
        return await this.widget?.locate();
    }
    get el() { return this; }
    static get style() { return ArcgisLocateStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-locate"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-locate":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisLocate$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisLocate = ArcgisLocate$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisLocate, defineCustomElement };
