// @arcgis/map-components/dist/components/arcgis-scale-bar.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { _ as newWidgetsScaleBar } from './index2.js';
import { u as useControllerManager } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';

const scaleBarCss = "arcgis-scale-bar{}";
const ArcgisScaleBarStyle0 = scaleBarCss;

const useScaleBarWidget = makeWidgetController(newWidgetsScaleBar);
const __cmpMeta = [0,"arcgis-scale-bar",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"barStyle":[1025,"bar-style"],"unit":[1025],"icon":[1025],"label":[1025],"position":[1025],"destroy":[64]}];
const ArcgisScaleBar$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisScaleBar extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useScaleBarWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.barStyle = this.widget.style;
        this.unit = this.widget.unit;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisScaleBarStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-scale-bar"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-scale-bar":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisScaleBar$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisScaleBar = ArcgisScaleBar$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisScaleBar, defineCustomElement };
