// @arcgis/map-components/dist/components/arcgis-basemap-toggle.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { o as newWidgetsBasemapToggle, h as importCoreReactiveUtils } from './index2.js';
import { l as load, u as usePropertyChange } from './chunk-PVEVZB4O.js';
import { u as useControllerManager } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';

const basemapToggleCss = "";
const ArcgisBasemapToggleStyle0 = basemapToggleCss;

const useBasemapToggleWidget = makeWidgetController(newWidgetsBasemapToggle);
const __cmpMeta = [0,"arcgis-basemap-toggle",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"position":[1025],"referenceElement":[1025,"reference-element"],"nextBasemap":[1040],"showTitle":[1028,"show-title"],"state":[1537],"destroy":[64],"toggle":[64]}];
const ArcgisBasemapToggle$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisBasemapToggle extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.reactiveUtils = load(importCoreReactiveUtils);
        this.widget = useBasemapToggleWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("nextBasemap", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.referenceElement = undefined;
        this.nextBasemap = this.widget.nextBasemap;
        this.showTitle = this.widget.visibleElements.title;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async toggle() {
        return await this.widget?.toggle();
    }
    get el() { return this; }
    static get style() { return ArcgisBasemapToggleStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-basemap-toggle"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-basemap-toggle":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisBasemapToggle$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisBasemapToggle = ArcgisBasemapToggle$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisBasemapToggle, defineCustomElement };
