// @arcgis/map-components/dist/components/arcgis-fullscreen.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { L as newWidgetsFullscreen } from './index2.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O.js';
import { u as useControllerManager } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';

const fullscreenCss = "";
const ArcgisFullscreenStyle0 = fullscreenCss;

const useFullscreenWidget = makeWidgetController(newWidgetsFullscreen);
const __cmpMeta = [0,"arcgis-fullscreen",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"icon":[1025],"label":[1025],"element":[1040],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisFullscreen$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisFullscreen extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useFullscreenWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.element = this.widget.element;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisFullscreenStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-fullscreen"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-fullscreen":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisFullscreen$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisFullscreen = ArcgisFullscreen$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisFullscreen, defineCustomElement };
