// @arcgis/map-components/dist/components/arcgis-compass.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { r as newWidgetsCompass } from './index2.js';
import { u as usePropertyChange } from './chunk-PVEVZB4O.js';
import { u as useControllerManager } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';

const compassCss = "";
const ArcgisCompassStyle0 = compassCss;

const useCompassWidget = makeWidgetController(newWidgetsCompass);
const __cmpMeta = [0,"arcgis-compass",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"orientation":[1040],"position":[1025],"referenceElement":[1025,"reference-element"],"state":[1537],"destroy":[64],"reset":[64]}];
const ArcgisCompass$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisCompass extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useCompassWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("orientation", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.orientation = this.viewModel.orientation;
        this.position = "bottom-left";
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async reset() {
        this.widget.reset();
    }
    get el() { return this; }
    static get style() { return ArcgisCompassStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-compass"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-compass":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisCompass$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisCompass = ArcgisCompass$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisCompass, defineCustomElement };
