// @arcgis/map-components/dist/components/version-management-version-item.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.7
 */
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import "@esri/calcite-components/dist/components/calcite-action.js";
import "@esri/calcite-components/dist/components/calcite-tooltip.js";
import "@esri/calcite-components/dist/components/calcite-list-item.js";
import "@esri/calcite-components/dist/components/calcite-action-menu.js";

const ArcgisVersionManagementVersionItem = /*@__PURE__*/ proxyCustomElement(class ArcgisVersionManagementVersionItem extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.arcgisManageVersion = createEvent(this, "arcgisManageVersion", 7);
        this.versionItemElementProps = undefined;
    }
    render() {
        const { versionItemElementProps: { currentUser, currentVersionIdentifier, hasAdvEditingUte, isVersioningApiAvailable, serviceUrl, strings: { actions }, versionInfo, }, } = this;
        const fullVersionName = versionInfo.versionIdentifier.name;
        const versionName = fullVersionName.substring(fullVersionName.lastIndexOf(".") + 1); // label
        const versionOwner = fullVersionName.substring(0, fullVersionName.lastIndexOf(".")).toUpperCase(); // description
        const guid = versionInfo.versionIdentifier.guid;
        const actionElementId = `actionSwitchToVersion_${guid}`;
        const canEditVersion = versionOwner.toUpperCase() === currentUser.toUpperCase() && isVersioningApiAvailable && hasAdvEditingUte;
        const isCurrentVersion = currentVersionIdentifier.guid === versionInfo.versionIdentifier.guid;
        const startActionChangeVersion = (h("calcite-action", { key: '2f6e3eeea7fc7736a1199a29668894dd80dac94a', disabled: isCurrentVersion, id: actionElementId, icon: "code-branch", text: "", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl, versionInfo });
            }, slot: "actions-start" }));
        const startActionChangeVersionTooltip = !isCurrentVersion ? (h("calcite-tooltip", { "overlay-positioning": "fixed", referenceElement: actionElementId }, h("span", null, actions.switchToVersion))) : undefined;
        return (h("calcite-list-item", { key: '74b8319a683737aaaf8cfc5f06147eaa401195a6', description: versionOwner, label: versionName, selected: false, value: guid }, startActionChangeVersion, startActionChangeVersionTooltip, h("calcite-action-menu", { key: '535784b416984e07bbfc44a275dfc5c81fb55daf', "overlay-positioning": "fixed", slot: "actions-end", label: "" }, h("calcite-action", { key: '5e89410767d43ab3ce8a6e3ac6ef5d2db5542ea7', disabled: isCurrentVersion, icon: "code-branch", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.switchToVersion, textEnabled: true }), h("calcite-action", { key: 'cf3ab694540fb02bcfa7c5ef9557fcb6a93746c1', disabled: !canEditVersion || isCurrentVersion, icon: "code-branch-edit", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "editVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.editVersion, textEnabled: true }), h("calcite-action", { key: 'bb646b907dbbc62283a15295a265912ba45d73a4', disabled: !canEditVersion || (canEditVersion && isCurrentVersion), icon: "trash", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "deleteVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.deleteVersion, textEnabled: true }))));
    }
    get el() { return this; }
}, [0, "arcgis-version-management-version-item", {
        "versionItemElementProps": [16]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-version-management-version-item"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-version-management-version-item":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisVersionManagementVersionItem);
            }
            break;
    } });
}
defineCustomElement();

export { ArcgisVersionManagementVersionItem as A, defineCustomElement as d };
