// timezone-groups/dist/utils/time-zones.mjs
var ctz;
(function (ctz) {
    ctz[ctz["Africa/Abidjan"] = 0] = "Africa/Abidjan";
    ctz[ctz["Africa/Algiers"] = 1] = "Africa/Algiers";
    ctz[ctz["Africa/Bissau"] = 2] = "Africa/Bissau";
    ctz[ctz["Africa/Cairo"] = 3] = "Africa/Cairo";
    ctz[ctz["Africa/Casablanca"] = 4] = "Africa/Casablanca";
    ctz[ctz["Africa/Ceuta"] = 5] = "Africa/Ceuta";
    ctz[ctz["Africa/El_Aaiun"] = 6] = "Africa/El_Aaiun";
    ctz[ctz["Africa/Johannesburg"] = 7] = "Africa/Johannesburg";
    ctz[ctz["Africa/Juba"] = 8] = "Africa/Juba";
    ctz[ctz["Africa/Khartoum"] = 9] = "Africa/Khartoum";
    ctz[ctz["Africa/Lagos"] = 10] = "Africa/Lagos";
    ctz[ctz["Africa/Maputo"] = 11] = "Africa/Maputo";
    ctz[ctz["Africa/Monrovia"] = 12] = "Africa/Monrovia";
    ctz[ctz["Africa/Nairobi"] = 13] = "Africa/Nairobi";
    ctz[ctz["Africa/Ndjamena"] = 14] = "Africa/Ndjamena";
    ctz[ctz["Africa/Sao_Tome"] = 15] = "Africa/Sao_Tome";
    ctz[ctz["Africa/Tripoli"] = 16] = "Africa/Tripoli";
    ctz[ctz["Africa/Tunis"] = 17] = "Africa/Tunis";
    ctz[ctz["Africa/Windhoek"] = 18] = "Africa/Windhoek";
    ctz[ctz["America/Adak"] = 19] = "America/Adak";
    ctz[ctz["America/Anchorage"] = 20] = "America/Anchorage";
    ctz[ctz["America/Araguaina"] = 21] = "America/Araguaina";
    ctz[ctz["America/Argentina/Buenos_Aires"] = 22] = "America/Argentina/Buenos_Aires";
    ctz[ctz["America/Argentina/Catamarca"] = 23] = "America/Argentina/Catamarca";
    ctz[ctz["America/Argentina/Cordoba"] = 24] = "America/Argentina/Cordoba";
    ctz[ctz["America/Argentina/Jujuy"] = 25] = "America/Argentina/Jujuy";
    ctz[ctz["America/Argentina/La_Rioja"] = 26] = "America/Argentina/La_Rioja";
    ctz[ctz["America/Argentina/Mendoza"] = 27] = "America/Argentina/Mendoza";
    ctz[ctz["America/Argentina/Rio_Gallegos"] = 28] = "America/Argentina/Rio_Gallegos";
    ctz[ctz["America/Argentina/Salta"] = 29] = "America/Argentina/Salta";
    ctz[ctz["America/Argentina/San_Juan"] = 30] = "America/Argentina/San_Juan";
    ctz[ctz["America/Argentina/San_Luis"] = 31] = "America/Argentina/San_Luis";
    ctz[ctz["America/Argentina/Tucuman"] = 32] = "America/Argentina/Tucuman";
    ctz[ctz["America/Argentina/Ushuaia"] = 33] = "America/Argentina/Ushuaia";
    ctz[ctz["America/Asuncion"] = 34] = "America/Asuncion";
    ctz[ctz["America/Bahia"] = 35] = "America/Bahia";
    ctz[ctz["America/Bahia_Banderas"] = 36] = "America/Bahia_Banderas";
    ctz[ctz["America/Barbados"] = 37] = "America/Barbados";
    ctz[ctz["America/Belem"] = 38] = "America/Belem";
    ctz[ctz["America/Belize"] = 39] = "America/Belize";
    ctz[ctz["America/Boa_Vista"] = 40] = "America/Boa_Vista";
    ctz[ctz["America/Bogota"] = 41] = "America/Bogota";
    ctz[ctz["America/Boise"] = 42] = "America/Boise";
    ctz[ctz["America/Cambridge_Bay"] = 43] = "America/Cambridge_Bay";
    ctz[ctz["America/Campo_Grande"] = 44] = "America/Campo_Grande";
    ctz[ctz["America/Cancun"] = 45] = "America/Cancun";
    ctz[ctz["America/Caracas"] = 46] = "America/Caracas";
    ctz[ctz["America/Cayenne"] = 47] = "America/Cayenne";
    ctz[ctz["America/Chicago"] = 48] = "America/Chicago";
    ctz[ctz["America/Chihuahua"] = 49] = "America/Chihuahua";
    ctz[ctz["America/Ciudad_Juarez"] = 50] = "America/Ciudad_Juarez";
    ctz[ctz["America/Costa_Rica"] = 51] = "America/Costa_Rica";
    ctz[ctz["America/Cuiaba"] = 52] = "America/Cuiaba";
    ctz[ctz["America/Danmarkshavn"] = 53] = "America/Danmarkshavn";
    ctz[ctz["America/Dawson"] = 54] = "America/Dawson";
    ctz[ctz["America/Dawson_Creek"] = 55] = "America/Dawson_Creek";
    ctz[ctz["America/Denver"] = 56] = "America/Denver";
    ctz[ctz["America/Detroit"] = 57] = "America/Detroit";
    ctz[ctz["America/Edmonton"] = 58] = "America/Edmonton";
    ctz[ctz["America/Eirunepe"] = 59] = "America/Eirunepe";
    ctz[ctz["America/El_Salvador"] = 60] = "America/El_Salvador";
    ctz[ctz["America/Fort_Nelson"] = 61] = "America/Fort_Nelson";
    ctz[ctz["America/Fortaleza"] = 62] = "America/Fortaleza";
    ctz[ctz["America/Glace_Bay"] = 63] = "America/Glace_Bay";
    ctz[ctz["America/Goose_Bay"] = 64] = "America/Goose_Bay";
    ctz[ctz["America/Grand_Turk"] = 65] = "America/Grand_Turk";
    ctz[ctz["America/Guatemala"] = 66] = "America/Guatemala";
    ctz[ctz["America/Guayaquil"] = 67] = "America/Guayaquil";
    ctz[ctz["America/Guyana"] = 68] = "America/Guyana";
    ctz[ctz["America/Halifax"] = 69] = "America/Halifax";
    ctz[ctz["America/Havana"] = 70] = "America/Havana";
    ctz[ctz["America/Hermosillo"] = 71] = "America/Hermosillo";
    ctz[ctz["America/Indiana/Indianapolis"] = 72] = "America/Indiana/Indianapolis";
    ctz[ctz["America/Indiana/Knox"] = 73] = "America/Indiana/Knox";
    ctz[ctz["America/Indiana/Marengo"] = 74] = "America/Indiana/Marengo";
    ctz[ctz["America/Indiana/Petersburg"] = 75] = "America/Indiana/Petersburg";
    ctz[ctz["America/Indiana/Tell_City"] = 76] = "America/Indiana/Tell_City";
    ctz[ctz["America/Indiana/Vevay"] = 77] = "America/Indiana/Vevay";
    ctz[ctz["America/Indiana/Vincennes"] = 78] = "America/Indiana/Vincennes";
    ctz[ctz["America/Indiana/Winamac"] = 79] = "America/Indiana/Winamac";
    ctz[ctz["America/Inuvik"] = 80] = "America/Inuvik";
    ctz[ctz["America/Iqaluit"] = 81] = "America/Iqaluit";
    ctz[ctz["America/Jamaica"] = 82] = "America/Jamaica";
    ctz[ctz["America/Juneau"] = 83] = "America/Juneau";
    ctz[ctz["America/Kentucky/Louisville"] = 84] = "America/Kentucky/Louisville";
    ctz[ctz["America/Kentucky/Monticello"] = 85] = "America/Kentucky/Monticello";
    ctz[ctz["America/La_Paz"] = 86] = "America/La_Paz";
    ctz[ctz["America/Lima"] = 87] = "America/Lima";
    ctz[ctz["America/Los_Angeles"] = 88] = "America/Los_Angeles";
    ctz[ctz["America/Maceio"] = 89] = "America/Maceio";
    ctz[ctz["America/Managua"] = 90] = "America/Managua";
    ctz[ctz["America/Manaus"] = 91] = "America/Manaus";
    ctz[ctz["America/Martinique"] = 92] = "America/Martinique";
    ctz[ctz["America/Matamoros"] = 93] = "America/Matamoros";
    ctz[ctz["America/Mazatlan"] = 94] = "America/Mazatlan";
    ctz[ctz["America/Menominee"] = 95] = "America/Menominee";
    ctz[ctz["America/Merida"] = 96] = "America/Merida";
    ctz[ctz["America/Metlakatla"] = 97] = "America/Metlakatla";
    ctz[ctz["America/Mexico_City"] = 98] = "America/Mexico_City";
    ctz[ctz["America/Miquelon"] = 99] = "America/Miquelon";
    ctz[ctz["America/Moncton"] = 100] = "America/Moncton";
    ctz[ctz["America/Monterrey"] = 101] = "America/Monterrey";
    ctz[ctz["America/Montevideo"] = 102] = "America/Montevideo";
    ctz[ctz["America/New_York"] = 103] = "America/New_York";
    ctz[ctz["America/Nome"] = 104] = "America/Nome";
    ctz[ctz["America/Noronha"] = 105] = "America/Noronha";
    ctz[ctz["America/North_Dakota/Beulah"] = 106] = "America/North_Dakota/Beulah";
    ctz[ctz["America/North_Dakota/Center"] = 107] = "America/North_Dakota/Center";
    ctz[ctz["America/North_Dakota/New_Salem"] = 108] = "America/North_Dakota/New_Salem";
    ctz[ctz["America/Nuuk"] = 109] = "America/Nuuk";
    ctz[ctz["America/Ojinaga"] = 110] = "America/Ojinaga";
    ctz[ctz["America/Panama"] = 111] = "America/Panama";
    ctz[ctz["America/Paramaribo"] = 112] = "America/Paramaribo";
    ctz[ctz["America/Phoenix"] = 113] = "America/Phoenix";
    ctz[ctz["America/Port-au-Prince"] = 114] = "America/Port-au-Prince";
    ctz[ctz["America/Porto_Velho"] = 115] = "America/Porto_Velho";
    ctz[ctz["America/Puerto_Rico"] = 116] = "America/Puerto_Rico";
    ctz[ctz["America/Punta_Arenas"] = 117] = "America/Punta_Arenas";
    ctz[ctz["America/Rankin_Inlet"] = 118] = "America/Rankin_Inlet";
    ctz[ctz["America/Recife"] = 119] = "America/Recife";
    ctz[ctz["America/Regina"] = 120] = "America/Regina";
    ctz[ctz["America/Resolute"] = 121] = "America/Resolute";
    ctz[ctz["America/Rio_Branco"] = 122] = "America/Rio_Branco";
    ctz[ctz["America/Santarem"] = 123] = "America/Santarem";
    ctz[ctz["America/Santiago"] = 124] = "America/Santiago";
    ctz[ctz["America/Santo_Domingo"] = 125] = "America/Santo_Domingo";
    ctz[ctz["America/Sao_Paulo"] = 126] = "America/Sao_Paulo";
    ctz[ctz["America/Scoresbysund"] = 127] = "America/Scoresbysund";
    ctz[ctz["America/Sitka"] = 128] = "America/Sitka";
    ctz[ctz["America/St_Johns"] = 129] = "America/St_Johns";
    ctz[ctz["America/Swift_Current"] = 130] = "America/Swift_Current";
    ctz[ctz["America/Tegucigalpa"] = 131] = "America/Tegucigalpa";
    ctz[ctz["America/Thule"] = 132] = "America/Thule";
    ctz[ctz["America/Tijuana"] = 133] = "America/Tijuana";
    ctz[ctz["America/Toronto"] = 134] = "America/Toronto";
    ctz[ctz["America/Vancouver"] = 135] = "America/Vancouver";
    ctz[ctz["America/Whitehorse"] = 136] = "America/Whitehorse";
    ctz[ctz["America/Winnipeg"] = 137] = "America/Winnipeg";
    ctz[ctz["America/Yakutat"] = 138] = "America/Yakutat";
    ctz[ctz["Antarctica/Casey"] = 139] = "Antarctica/Casey";
    ctz[ctz["Antarctica/Davis"] = 140] = "Antarctica/Davis";
    ctz[ctz["Antarctica/Macquarie"] = 141] = "Antarctica/Macquarie";
    ctz[ctz["Antarctica/Mawson"] = 142] = "Antarctica/Mawson";
    ctz[ctz["Antarctica/Palmer"] = 143] = "Antarctica/Palmer";
    ctz[ctz["Antarctica/Rothera"] = 144] = "Antarctica/Rothera";
    ctz[ctz["Antarctica/Troll"] = 145] = "Antarctica/Troll";
    ctz[ctz["Antarctica/Vostok"] = 146] = "Antarctica/Vostok";
    ctz[ctz["Asia/Almaty"] = 147] = "Asia/Almaty";
    ctz[ctz["Asia/Amman"] = 148] = "Asia/Amman";
    ctz[ctz["Asia/Anadyr"] = 149] = "Asia/Anadyr";
    ctz[ctz["Asia/Aqtau"] = 150] = "Asia/Aqtau";
    ctz[ctz["Asia/Aqtobe"] = 151] = "Asia/Aqtobe";
    ctz[ctz["Asia/Ashgabat"] = 152] = "Asia/Ashgabat";
    ctz[ctz["Asia/Atyrau"] = 153] = "Asia/Atyrau";
    ctz[ctz["Asia/Baghdad"] = 154] = "Asia/Baghdad";
    ctz[ctz["Asia/Baku"] = 155] = "Asia/Baku";
    ctz[ctz["Asia/Bangkok"] = 156] = "Asia/Bangkok";
    ctz[ctz["Asia/Barnaul"] = 157] = "Asia/Barnaul";
    ctz[ctz["Asia/Beirut"] = 158] = "Asia/Beirut";
    ctz[ctz["Asia/Bishkek"] = 159] = "Asia/Bishkek";
    ctz[ctz["Asia/Chita"] = 160] = "Asia/Chita";
    ctz[ctz["Asia/Choibalsan"] = 161] = "Asia/Choibalsan";
    ctz[ctz["Asia/Colombo"] = 162] = "Asia/Colombo";
    ctz[ctz["Asia/Damascus"] = 163] = "Asia/Damascus";
    ctz[ctz["Asia/Dhaka"] = 164] = "Asia/Dhaka";
    ctz[ctz["Asia/Dili"] = 165] = "Asia/Dili";
    ctz[ctz["Asia/Dubai"] = 166] = "Asia/Dubai";
    ctz[ctz["Asia/Dushanbe"] = 167] = "Asia/Dushanbe";
    ctz[ctz["Asia/Famagusta"] = 168] = "Asia/Famagusta";
    ctz[ctz["Asia/Gaza"] = 169] = "Asia/Gaza";
    ctz[ctz["Asia/Hebron"] = 170] = "Asia/Hebron";
    ctz[ctz["Asia/Ho_Chi_Minh"] = 171] = "Asia/Ho_Chi_Minh";
    ctz[ctz["Asia/Hong_Kong"] = 172] = "Asia/Hong_Kong";
    ctz[ctz["Asia/Hovd"] = 173] = "Asia/Hovd";
    ctz[ctz["Asia/Irkutsk"] = 174] = "Asia/Irkutsk";
    ctz[ctz["Asia/Jakarta"] = 175] = "Asia/Jakarta";
    ctz[ctz["Asia/Jayapura"] = 176] = "Asia/Jayapura";
    ctz[ctz["Asia/Jerusalem"] = 177] = "Asia/Jerusalem";
    ctz[ctz["Asia/Kabul"] = 178] = "Asia/Kabul";
    ctz[ctz["Asia/Kamchatka"] = 179] = "Asia/Kamchatka";
    ctz[ctz["Asia/Karachi"] = 180] = "Asia/Karachi";
    ctz[ctz["Asia/Kathmandu"] = 181] = "Asia/Kathmandu";
    ctz[ctz["Asia/Khandyga"] = 182] = "Asia/Khandyga";
    ctz[ctz["Asia/Kolkata"] = 183] = "Asia/Kolkata";
    ctz[ctz["Asia/Krasnoyarsk"] = 184] = "Asia/Krasnoyarsk";
    ctz[ctz["Asia/Kuching"] = 185] = "Asia/Kuching";
    ctz[ctz["Asia/Macau"] = 186] = "Asia/Macau";
    ctz[ctz["Asia/Magadan"] = 187] = "Asia/Magadan";
    ctz[ctz["Asia/Makassar"] = 188] = "Asia/Makassar";
    ctz[ctz["Asia/Manila"] = 189] = "Asia/Manila";
    ctz[ctz["Asia/Nicosia"] = 190] = "Asia/Nicosia";
    ctz[ctz["Asia/Novokuznetsk"] = 191] = "Asia/Novokuznetsk";
    ctz[ctz["Asia/Novosibirsk"] = 192] = "Asia/Novosibirsk";
    ctz[ctz["Asia/Omsk"] = 193] = "Asia/Omsk";
    ctz[ctz["Asia/Oral"] = 194] = "Asia/Oral";
    ctz[ctz["Asia/Pontianak"] = 195] = "Asia/Pontianak";
    ctz[ctz["Asia/Pyongyang"] = 196] = "Asia/Pyongyang";
    ctz[ctz["Asia/Qatar"] = 197] = "Asia/Qatar";
    ctz[ctz["Asia/Qostanay"] = 198] = "Asia/Qostanay";
    ctz[ctz["Asia/Qyzylorda"] = 199] = "Asia/Qyzylorda";
    ctz[ctz["Asia/Riyadh"] = 200] = "Asia/Riyadh";
    ctz[ctz["Asia/Sakhalin"] = 201] = "Asia/Sakhalin";
    ctz[ctz["Asia/Samarkand"] = 202] = "Asia/Samarkand";
    ctz[ctz["Asia/Seoul"] = 203] = "Asia/Seoul";
    ctz[ctz["Asia/Shanghai"] = 204] = "Asia/Shanghai";
    ctz[ctz["Asia/Singapore"] = 205] = "Asia/Singapore";
    ctz[ctz["Asia/Srednekolymsk"] = 206] = "Asia/Srednekolymsk";
    ctz[ctz["Asia/Taipei"] = 207] = "Asia/Taipei";
    ctz[ctz["Asia/Tashkent"] = 208] = "Asia/Tashkent";
    ctz[ctz["Asia/Tbilisi"] = 209] = "Asia/Tbilisi";
    ctz[ctz["Asia/Tehran"] = 210] = "Asia/Tehran";
    ctz[ctz["Asia/Thimphu"] = 211] = "Asia/Thimphu";
    ctz[ctz["Asia/Tokyo"] = 212] = "Asia/Tokyo";
    ctz[ctz["Asia/Tomsk"] = 213] = "Asia/Tomsk";
    ctz[ctz["Asia/Ulaanbaatar"] = 214] = "Asia/Ulaanbaatar";
    ctz[ctz["Asia/Urumqi"] = 215] = "Asia/Urumqi";
    ctz[ctz["Asia/Ust-Nera"] = 216] = "Asia/Ust-Nera";
    ctz[ctz["Asia/Vladivostok"] = 217] = "Asia/Vladivostok";
    ctz[ctz["Asia/Yakutsk"] = 218] = "Asia/Yakutsk";
    ctz[ctz["Asia/Yangon"] = 219] = "Asia/Yangon";
    ctz[ctz["Asia/Yekaterinburg"] = 220] = "Asia/Yekaterinburg";
    ctz[ctz["Asia/Yerevan"] = 221] = "Asia/Yerevan";
    ctz[ctz["Atlantic/Azores"] = 222] = "Atlantic/Azores";
    ctz[ctz["Atlantic/Bermuda"] = 223] = "Atlantic/Bermuda";
    ctz[ctz["Atlantic/Canary"] = 224] = "Atlantic/Canary";
    ctz[ctz["Atlantic/Cape_Verde"] = 225] = "Atlantic/Cape_Verde";
    ctz[ctz["Atlantic/Faroe"] = 226] = "Atlantic/Faroe";
    ctz[ctz["Atlantic/Madeira"] = 227] = "Atlantic/Madeira";
    ctz[ctz["Atlantic/South_Georgia"] = 228] = "Atlantic/South_Georgia";
    ctz[ctz["Atlantic/Stanley"] = 229] = "Atlantic/Stanley";
    ctz[ctz["Australia/Adelaide"] = 230] = "Australia/Adelaide";
    ctz[ctz["Australia/Brisbane"] = 231] = "Australia/Brisbane";
    ctz[ctz["Australia/Broken_Hill"] = 232] = "Australia/Broken_Hill";
    ctz[ctz["Australia/Darwin"] = 233] = "Australia/Darwin";
    ctz[ctz["Australia/Eucla"] = 234] = "Australia/Eucla";
    ctz[ctz["Australia/Hobart"] = 235] = "Australia/Hobart";
    ctz[ctz["Australia/Lindeman"] = 236] = "Australia/Lindeman";
    ctz[ctz["Australia/Lord_Howe"] = 237] = "Australia/Lord_Howe";
    ctz[ctz["Australia/Melbourne"] = 238] = "Australia/Melbourne";
    ctz[ctz["Australia/Perth"] = 239] = "Australia/Perth";
    ctz[ctz["Australia/Sydney"] = 240] = "Australia/Sydney";
    ctz[ctz["CET"] = 241] = "CET";
    ctz[ctz["CST6CDT"] = 242] = "CST6CDT";
    ctz[ctz["EET"] = 243] = "EET";
    ctz[ctz["EST"] = 244] = "EST";
    ctz[ctz["EST5EDT"] = 245] = "EST5EDT";
    ctz[ctz["Etc/GMT"] = 246] = "Etc/GMT";
    ctz[ctz["Etc/GMT+1"] = 247] = "Etc/GMT+1";
    ctz[ctz["Etc/GMT+10"] = 248] = "Etc/GMT+10";
    ctz[ctz["Etc/GMT+11"] = 249] = "Etc/GMT+11";
    ctz[ctz["Etc/GMT+12"] = 250] = "Etc/GMT+12";
    ctz[ctz["Etc/GMT+2"] = 251] = "Etc/GMT+2";
    ctz[ctz["Etc/GMT+3"] = 252] = "Etc/GMT+3";
    ctz[ctz["Etc/GMT+4"] = 253] = "Etc/GMT+4";
    ctz[ctz["Etc/GMT+5"] = 254] = "Etc/GMT+5";
    ctz[ctz["Etc/GMT+6"] = 255] = "Etc/GMT+6";
    ctz[ctz["Etc/GMT+7"] = 256] = "Etc/GMT+7";
    ctz[ctz["Etc/GMT+8"] = 257] = "Etc/GMT+8";
    ctz[ctz["Etc/GMT+9"] = 258] = "Etc/GMT+9";
    ctz[ctz["Etc/GMT-1"] = 259] = "Etc/GMT-1";
    ctz[ctz["Etc/GMT-10"] = 260] = "Etc/GMT-10";
    ctz[ctz["Etc/GMT-11"] = 261] = "Etc/GMT-11";
    ctz[ctz["Etc/GMT-12"] = 262] = "Etc/GMT-12";
    ctz[ctz["Etc/GMT-13"] = 263] = "Etc/GMT-13";
    ctz[ctz["Etc/GMT-14"] = 264] = "Etc/GMT-14";
    ctz[ctz["Etc/GMT-2"] = 265] = "Etc/GMT-2";
    ctz[ctz["Etc/GMT-3"] = 266] = "Etc/GMT-3";
    ctz[ctz["Etc/GMT-4"] = 267] = "Etc/GMT-4";
    ctz[ctz["Etc/GMT-5"] = 268] = "Etc/GMT-5";
    ctz[ctz["Etc/GMT-6"] = 269] = "Etc/GMT-6";
    ctz[ctz["Etc/GMT-7"] = 270] = "Etc/GMT-7";
    ctz[ctz["Etc/GMT-8"] = 271] = "Etc/GMT-8";
    ctz[ctz["Etc/GMT-9"] = 272] = "Etc/GMT-9";
    ctz[ctz["Etc/UTC"] = 273] = "Etc/UTC";
    ctz[ctz["Europe/Andorra"] = 274] = "Europe/Andorra";
    ctz[ctz["Europe/Astrakhan"] = 275] = "Europe/Astrakhan";
    ctz[ctz["Europe/Athens"] = 276] = "Europe/Athens";
    ctz[ctz["Europe/Belgrade"] = 277] = "Europe/Belgrade";
    ctz[ctz["Europe/Berlin"] = 278] = "Europe/Berlin";
    ctz[ctz["Europe/Brussels"] = 279] = "Europe/Brussels";
    ctz[ctz["Europe/Bucharest"] = 280] = "Europe/Bucharest";
    ctz[ctz["Europe/Budapest"] = 281] = "Europe/Budapest";
    ctz[ctz["Europe/Chisinau"] = 282] = "Europe/Chisinau";
    ctz[ctz["Europe/Dublin"] = 283] = "Europe/Dublin";
    ctz[ctz["Europe/Gibraltar"] = 284] = "Europe/Gibraltar";
    ctz[ctz["Europe/Helsinki"] = 285] = "Europe/Helsinki";
    ctz[ctz["Europe/Istanbul"] = 286] = "Europe/Istanbul";
    ctz[ctz["Europe/Kaliningrad"] = 287] = "Europe/Kaliningrad";
    ctz[ctz["Europe/Kirov"] = 288] = "Europe/Kirov";
    ctz[ctz["Europe/Kyiv"] = 289] = "Europe/Kyiv";
    ctz[ctz["Europe/Lisbon"] = 290] = "Europe/Lisbon";
    ctz[ctz["Europe/London"] = 291] = "Europe/London";
    ctz[ctz["Europe/Madrid"] = 292] = "Europe/Madrid";
    ctz[ctz["Europe/Malta"] = 293] = "Europe/Malta";
    ctz[ctz["Europe/Minsk"] = 294] = "Europe/Minsk";
    ctz[ctz["Europe/Moscow"] = 295] = "Europe/Moscow";
    ctz[ctz["Europe/Paris"] = 296] = "Europe/Paris";
    ctz[ctz["Europe/Prague"] = 297] = "Europe/Prague";
    ctz[ctz["Europe/Riga"] = 298] = "Europe/Riga";
    ctz[ctz["Europe/Rome"] = 299] = "Europe/Rome";
    ctz[ctz["Europe/Samara"] = 300] = "Europe/Samara";
    ctz[ctz["Europe/Saratov"] = 301] = "Europe/Saratov";
    ctz[ctz["Europe/Simferopol"] = 302] = "Europe/Simferopol";
    ctz[ctz["Europe/Sofia"] = 303] = "Europe/Sofia";
    ctz[ctz["Europe/Tallinn"] = 304] = "Europe/Tallinn";
    ctz[ctz["Europe/Tirane"] = 305] = "Europe/Tirane";
    ctz[ctz["Europe/Ulyanovsk"] = 306] = "Europe/Ulyanovsk";
    ctz[ctz["Europe/Vienna"] = 307] = "Europe/Vienna";
    ctz[ctz["Europe/Vilnius"] = 308] = "Europe/Vilnius";
    ctz[ctz["Europe/Volgograd"] = 309] = "Europe/Volgograd";
    ctz[ctz["Europe/Warsaw"] = 310] = "Europe/Warsaw";
    ctz[ctz["Europe/Zurich"] = 311] = "Europe/Zurich";
    ctz[ctz["Factory"] = 312] = "Factory";
    ctz[ctz["HST"] = 313] = "HST";
    ctz[ctz["Indian/Chagos"] = 314] = "Indian/Chagos";
    ctz[ctz["Indian/Maldives"] = 315] = "Indian/Maldives";
    ctz[ctz["Indian/Mauritius"] = 316] = "Indian/Mauritius";
    ctz[ctz["MET"] = 317] = "MET";
    ctz[ctz["MST"] = 318] = "MST";
    ctz[ctz["MST7MDT"] = 319] = "MST7MDT";
    ctz[ctz["Pacific/Apia"] = 320] = "Pacific/Apia";
    ctz[ctz["Pacific/Auckland"] = 321] = "Pacific/Auckland";
    ctz[ctz["Pacific/Bougainville"] = 322] = "Pacific/Bougainville";
    ctz[ctz["Pacific/Chatham"] = 323] = "Pacific/Chatham";
    ctz[ctz["Pacific/Easter"] = 324] = "Pacific/Easter";
    ctz[ctz["Pacific/Efate"] = 325] = "Pacific/Efate";
    ctz[ctz["Pacific/Fakaofo"] = 326] = "Pacific/Fakaofo";
    ctz[ctz["Pacific/Fiji"] = 327] = "Pacific/Fiji";
    ctz[ctz["Pacific/Galapagos"] = 328] = "Pacific/Galapagos";
    ctz[ctz["Pacific/Gambier"] = 329] = "Pacific/Gambier";
    ctz[ctz["Pacific/Guadalcanal"] = 330] = "Pacific/Guadalcanal";
    ctz[ctz["Pacific/Guam"] = 331] = "Pacific/Guam";
    ctz[ctz["Pacific/Honolulu"] = 332] = "Pacific/Honolulu";
    ctz[ctz["Pacific/Kanton"] = 333] = "Pacific/Kanton";
    ctz[ctz["Pacific/Kiritimati"] = 334] = "Pacific/Kiritimati";
    ctz[ctz["Pacific/Kosrae"] = 335] = "Pacific/Kosrae";
    ctz[ctz["Pacific/Kwajalein"] = 336] = "Pacific/Kwajalein";
    ctz[ctz["Pacific/Marquesas"] = 337] = "Pacific/Marquesas";
    ctz[ctz["Pacific/Nauru"] = 338] = "Pacific/Nauru";
    ctz[ctz["Pacific/Niue"] = 339] = "Pacific/Niue";
    ctz[ctz["Pacific/Norfolk"] = 340] = "Pacific/Norfolk";
    ctz[ctz["Pacific/Noumea"] = 341] = "Pacific/Noumea";
    ctz[ctz["Pacific/Pago_Pago"] = 342] = "Pacific/Pago_Pago";
    ctz[ctz["Pacific/Palau"] = 343] = "Pacific/Palau";
    ctz[ctz["Pacific/Pitcairn"] = 344] = "Pacific/Pitcairn";
    ctz[ctz["Pacific/Port_Moresby"] = 345] = "Pacific/Port_Moresby";
    ctz[ctz["Pacific/Rarotonga"] = 346] = "Pacific/Rarotonga";
    ctz[ctz["Pacific/Tahiti"] = 347] = "Pacific/Tahiti";
    ctz[ctz["Pacific/Tarawa"] = 348] = "Pacific/Tarawa";
    ctz[ctz["Pacific/Tongatapu"] = 349] = "Pacific/Tongatapu";
    ctz[ctz["PST8PDT"] = 350] = "PST8PDT";
    ctz[ctz["WET"] = 351] = "WET";
})(ctz || (ctz = {}));
const timeZones = (() => {
    return Object.values(ctz);
})();
const linkedTimeZones = {
    "Africa/Asmera": ctz["Africa/Nairobi"],
    "America/Atka": ctz["America/Adak"],
    "America/Buenos_Aires": ctz["America/Argentina/Buenos_Aires"],
    "America/Catamarca": ctz["America/Argentina/Catamarca"],
    "America/Cordoba": ctz["America/Argentina/Cordoba"],
    "America/Fort_Wayne": ctz["America/Indiana/Indianapolis"],
    "America/Godthab": ctz["America/Nuuk"],
    "America/Indianapolis": ctz["America/Indiana/Indianapolis"],
    "America/Jujuy": ctz["America/Argentina/Jujuy"],
    "America/Knox_IN": ctz["America/Indiana/Knox"],
    "America/Kralendijk": ctz["America/Puerto_Rico"],
    "America/Louisville": ctz["America/Kentucky/Louisville"],
    "America/Lower_Princes": ctz["America/Puerto_Rico"],
    "America/Marigot": ctz["America/Puerto_Rico"],
    "America/Mendoza": ctz["America/Argentina/Mendoza"],
    "America/Porto_Acre": ctz["America/Rio_Branco"],
    "America/Santa_Isabel": ctz["America/Tijuana"],
    "America/Shiprock": ctz["America/Denver"],
    "America/St_Barthelemy": ctz["America/Puerto_Rico"],
    "America/Virgin": ctz["America/Puerto_Rico"],
    "Antarctica/South_Pole": ctz["Pacific/Auckland"],
    "Arctic/Longyearbyen": ctz["Europe/Berlin"],
    "Asia/Ashkhabad": ctz["Asia/Ashgabat"],
    "Asia/Calcutta": ctz["Asia/Kolkata"],
    "Asia/Chungking": ctz["Asia/Shanghai"],
    "Asia/Dacca": ctz["Asia/Dhaka"],
    "Asia/Istanbul": ctz["Europe/Istanbul"],
    "Asia/Katmandu": ctz["Asia/Kathmandu"],
    "Asia/Macao": ctz["Asia/Macau"],
    "Asia/Rangoon": ctz["Asia/Yangon"],
    "Asia/Saigon": ctz["Asia/Ho_Chi_Minh"],
    "Asia/Thimbu": ctz["Asia/Thimphu"],
    "Asia/Ujung_Pandang": ctz["Asia/Makassar"],
    "Asia/Ulan_Bator": ctz["Asia/Ulaanbaatar"],
    "Atlantic/Faeroe": ctz["Atlantic/Faroe"],
    "Australia/ACT": ctz["Australia/Sydney"],
    "Australia/Canberra": ctz["Australia/Sydney"],
    "Australia/LHI": ctz["Australia/Lord_Howe"],
    "Australia/North": ctz["Australia/Darwin"],
    "Australia/NSW": ctz["Australia/Sydney"],
    "Australia/Queensland": ctz["Australia/Brisbane"],
    "Australia/South": ctz["Australia/Adelaide"],
    "Australia/Tasmania": ctz["Australia/Hobart"],
    "Australia/Victoria": ctz["Australia/Melbourne"],
    "Australia/West": ctz["Australia/Perth"],
    "Australia/Yancowinna": ctz["Australia/Broken_Hill"],
    "Brazil/Acre": ctz["America/Rio_Branco"],
    "Brazil/DeNoronha": ctz["America/Noronha"],
    "Brazil/East": ctz["America/Sao_Paulo"],
    "Brazil/West": ctz["America/Manaus"],
    "Canada/Atlantic": ctz["America/Halifax"],
    "Canada/Central": ctz["America/Winnipeg"],
    "Canada/Eastern": ctz["America/Toronto"],
    "Canada/Mountain": ctz["America/Edmonton"],
    "Canada/Newfoundland": ctz["America/St_Johns"],
    "Canada/Pacific": ctz["America/Vancouver"],
    "Canada/Saskatchewan": ctz["America/Regina"],
    "Canada/Yukon": ctz["America/Whitehorse"],
    "Chile/Continental": ctz["America/Santiago"],
    "Chile/EasterIsland": ctz["Pacific/Easter"],
    "Cuba": ctz["America/Havana"],
    "Egypt": ctz["Africa/Cairo"],
    "Eire": ctz["Europe/Dublin"],
    "Etc/GMT+0": ctz["Etc/GMT"],
    "Etc/GMT-0": ctz["Etc/GMT"],
    "Etc/GMT0": ctz["Etc/GMT"],
    "Etc/Greenwich": ctz["Etc/GMT"],
    "Etc/UCT": ctz["Etc/UTC"],
    "Etc/Universal": ctz["Etc/UTC"],
    "Etc/Zulu": ctz["Etc/UTC"],
    "Europe/Bratislava": ctz["Europe/Prague"],
    "Europe/Busingen": ctz["Europe/Zurich"],
    "Europe/Kiev": ctz["Europe/Kyiv"],
    "Europe/Mariehamn": ctz["Europe/Helsinki"],
    "Europe/Nicosia": ctz["Asia/Nicosia"],
    "Europe/Podgorica": ctz["Europe/Belgrade"],
    "Europe/San_Marino": ctz["Europe/Rome"],
    "Europe/Vatican": ctz["Europe/Rome"],
    "GB": ctz["Europe/London"],
    "GB-Eire": ctz["Europe/London"],
    "GMT": ctz["Etc/GMT"],
    "GMT+0": ctz["Etc/GMT"],
    "GMT-0": ctz["Etc/GMT"],
    "GMT0": ctz["Etc/GMT"],
    "Greenwich": ctz["Etc/GMT"],
    "Hongkong": ctz["Asia/Hong_Kong"],
    "Iceland": ctz["Africa/Abidjan"],
    "Iran": ctz["Asia/Tehran"],
    "Israel": ctz["Asia/Jerusalem"],
    "Jamaica": ctz["America/Jamaica"],
    "Japan": ctz["Asia/Tokyo"],
    "Kwajalein": ctz["Pacific/Kwajalein"],
    "Libya": ctz["Africa/Tripoli"],
    "Mexico/BajaNorte": ctz["America/Tijuana"],
    "Mexico/BajaSur": ctz["America/Mazatlan"],
    "Mexico/General": ctz["America/Mexico_City"],
    "Navajo": ctz["America/Denver"],
    "NZ": ctz["Pacific/Auckland"],
    "NZ-CHAT": ctz["Pacific/Chatham"],
    "Pacific/Ponape": ctz["Pacific/Guadalcanal"],
    "Pacific/Samoa": ctz["Pacific/Pago_Pago"],
    "Pacific/Truk": ctz["Pacific/Port_Moresby"],
    "Pacific/Yap": ctz["Pacific/Port_Moresby"],
    "Poland": ctz["Europe/Warsaw"],
    "Portugal": ctz["Europe/Lisbon"],
    "PRC": ctz["Asia/Shanghai"],
    "ROC": ctz["Asia/Taipei"],
    "ROK": ctz["Asia/Seoul"],
    "Singapore": ctz["Asia/Singapore"],
    "Turkey": ctz["Europe/Istanbul"],
    "UCT": ctz["Etc/UTC"],
    "Universal": ctz["Etc/UTC"],
    "US/Alaska": ctz["America/Anchorage"],
    "US/Aleutian": ctz["America/Adak"],
    "US/Arizona": ctz["America/Phoenix"],
    "US/Central": ctz["America/Chicago"],
    "US/East-Indiana": ctz["America/Indiana/Indianapolis"],
    "US/Eastern": ctz["America/New_York"],
    "US/Hawaii": ctz["Pacific/Honolulu"],
    "US/Indiana-Starke": ctz["America/Indiana/Knox"],
    "US/Michigan": ctz["America/Detroit"],
    "US/Mountain": ctz["America/Denver"],
    "US/Pacific": ctz["America/Los_Angeles"],
    "US/Samoa": ctz["Pacific/Pago_Pago"],
    "UTC": ctz["Etc/UTC"],
    "W-SU": ctz["Europe/Moscow"],
    "Zulu": ctz["Etc/UTC"],
    "Africa/Accra": ctz["Africa/Abidjan"],
    "Africa/Addis_Ababa": ctz["Africa/Nairobi"],
    "Africa/Asmara": ctz["Africa/Nairobi"],
    "Africa/Bamako": ctz["Africa/Abidjan"],
    "Africa/Bangui": ctz["Africa/Lagos"],
    "Africa/Banjul": ctz["Africa/Abidjan"],
    "Africa/Blantyre": ctz["Africa/Maputo"],
    "Africa/Brazzaville": ctz["Africa/Lagos"],
    "Africa/Bujumbura": ctz["Africa/Maputo"],
    "Africa/Conakry": ctz["Africa/Abidjan"],
    "Africa/Dakar": ctz["Africa/Abidjan"],
    "Africa/Dar_es_Salaam": ctz["Africa/Nairobi"],
    "Africa/Djibouti": ctz["Africa/Nairobi"],
    "Africa/Douala": ctz["Africa/Lagos"],
    "Africa/Freetown": ctz["Africa/Abidjan"],
    "Africa/Gaborone": ctz["Africa/Maputo"],
    "Africa/Harare": ctz["Africa/Maputo"],
    "Africa/Kampala": ctz["Africa/Nairobi"],
    "Africa/Kigali": ctz["Africa/Maputo"],
    "Africa/Kinshasa": ctz["Africa/Lagos"],
    "Africa/Libreville": ctz["Africa/Lagos"],
    "Africa/Lome": ctz["Africa/Abidjan"],
    "Africa/Luanda": ctz["Africa/Lagos"],
    "Africa/Lubumbashi": ctz["Africa/Maputo"],
    "Africa/Lusaka": ctz["Africa/Maputo"],
    "Africa/Malabo": ctz["Africa/Lagos"],
    "Africa/Maseru": ctz["Africa/Johannesburg"],
    "Africa/Mbabane": ctz["Africa/Johannesburg"],
    "Africa/Mogadishu": ctz["Africa/Nairobi"],
    "Africa/Niamey": ctz["Africa/Lagos"],
    "Africa/Nouakchott": ctz["Africa/Abidjan"],
    "Africa/Ouagadougou": ctz["Africa/Abidjan"],
    "Africa/Porto-Novo": ctz["Africa/Lagos"],
    "Africa/Timbuktu": ctz["Africa/Abidjan"],
    "America/Anguilla": ctz["America/Puerto_Rico"],
    "America/Antigua": ctz["America/Puerto_Rico"],
    "America/Argentina/ComodRivadavia": ctz["America/Argentina/Catamarca"],
    "America/Aruba": ctz["America/Puerto_Rico"],
    "America/Atikokan": ctz["America/Panama"],
    "America/Blanc-Sablon": ctz["America/Puerto_Rico"],
    "America/Cayman": ctz["America/Panama"],
    "America/Coral_Harbour": ctz["America/Panama"],
    "America/Creston": ctz["America/Phoenix"],
    "America/Curacao": ctz["America/Puerto_Rico"],
    "America/Dominica": ctz["America/Puerto_Rico"],
    "America/Ensenada": ctz["America/Tijuana"],
    "America/Grenada": ctz["America/Puerto_Rico"],
    "America/Guadeloupe": ctz["America/Puerto_Rico"],
    "America/Montreal": ctz["America/Toronto"],
    "America/Montserrat": ctz["America/Puerto_Rico"],
    "America/Nassau": ctz["America/Toronto"],
    "America/Nipigon": ctz["America/Toronto"],
    "America/Pangnirtung": ctz["America/Iqaluit"],
    "America/Port_of_Spain": ctz["America/Puerto_Rico"],
    "America/Rainy_River": ctz["America/Winnipeg"],
    "America/Rosario": ctz["America/Argentina/Cordoba"],
    "America/St_Kitts": ctz["America/Puerto_Rico"],
    "America/St_Lucia": ctz["America/Puerto_Rico"],
    "America/St_Thomas": ctz["America/Puerto_Rico"],
    "America/St_Vincent": ctz["America/Puerto_Rico"],
    "America/Thunder_Bay": ctz["America/Toronto"],
    "America/Tortola": ctz["America/Puerto_Rico"],
    "America/Yellowknife": ctz["America/Edmonton"],
    "Antarctica/DumontDUrville": ctz["Pacific/Port_Moresby"],
    "Antarctica/McMurdo": ctz["Pacific/Auckland"],
    "Antarctica/Syowa": ctz["Asia/Riyadh"],
    "Asia/Aden": ctz["Asia/Riyadh"],
    "Asia/Bahrain": ctz["Asia/Qatar"],
    "Asia/Brunei": ctz["Asia/Kuching"],
    "Asia/Chongqing": ctz["Asia/Shanghai"],
    "Asia/Harbin": ctz["Asia/Shanghai"],
    "Asia/Kashgar": ctz["Asia/Urumqi"],
    "Asia/Kuala_Lumpur": ctz["Asia/Singapore"],
    "Asia/Kuwait": ctz["Asia/Riyadh"],
    "Asia/Muscat": ctz["Asia/Dubai"],
    "Asia/Phnom_Penh": ctz["Asia/Bangkok"],
    "Asia/Tel_Aviv": ctz["Asia/Jerusalem"],
    "Asia/Vientiane": ctz["Asia/Bangkok"],
    "Atlantic/Jan_Mayen": ctz["Europe/Berlin"],
    "Atlantic/Reykjavik": ctz["Africa/Abidjan"],
    "Atlantic/St_Helena": ctz["Africa/Abidjan"],
    "Australia/Currie": ctz["Australia/Hobart"],
    "Europe/Amsterdam": ctz["Europe/Brussels"],
    "Europe/Belfast": ctz["Europe/London"],
    "Europe/Copenhagen": ctz["Europe/Berlin"],
    "Europe/Guernsey": ctz["Europe/London"],
    "Europe/Isle_of_Man": ctz["Europe/London"],
    "Europe/Jersey": ctz["Europe/London"],
    "Europe/Ljubljana": ctz["Europe/Belgrade"],
    "Europe/Luxembourg": ctz["Europe/Brussels"],
    "Europe/Monaco": ctz["Europe/Paris"],
    "Europe/Oslo": ctz["Europe/Berlin"],
    "Europe/Sarajevo": ctz["Europe/Belgrade"],
    "Europe/Skopje": ctz["Europe/Belgrade"],
    "Europe/Stockholm": ctz["Europe/Berlin"],
    "Europe/Tiraspol": ctz["Europe/Chisinau"],
    "Europe/Uzhgorod": ctz["Europe/Kyiv"],
    "Europe/Vaduz": ctz["Europe/Zurich"],
    "Europe/Zagreb": ctz["Europe/Belgrade"],
    "Europe/Zaporozhye": ctz["Europe/Kyiv"],
    "Indian/Antananarivo": ctz["Africa/Nairobi"],
    "Indian/Christmas": ctz["Asia/Bangkok"],
    "Indian/Cocos": ctz["Asia/Yangon"],
    "Indian/Comoro": ctz["Africa/Nairobi"],
    "Indian/Kerguelen": ctz["Indian/Maldives"],
    "Indian/Mahe": ctz["Asia/Dubai"],
    "Indian/Mayotte": ctz["Africa/Nairobi"],
    "Indian/Reunion": ctz["Asia/Dubai"],
    "Pacific/Chuuk": ctz["Pacific/Port_Moresby"],
    "Pacific/Enderbury": ctz["Pacific/Kanton"],
    "Pacific/Funafuti": ctz["Pacific/Tarawa"],
    "Pacific/Johnston": ctz["Pacific/Honolulu"],
    "Pacific/Majuro": ctz["Pacific/Tarawa"],
    "Pacific/Midway": ctz["Pacific/Pago_Pago"],
    "Pacific/Pohnpei": ctz["Pacific/Guadalcanal"],
    "Pacific/Saipan": ctz["Pacific/Guam"],
    "Pacific/Wake": ctz["Pacific/Tarawa"],
    "Pacific/Wallis": ctz["Pacific/Tarawa"]
};

export { linkedTimeZones, timeZones };
