// @esri/calcite-components/dist/cjs/ja-ad314678.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.12.1
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-6a00a289.js');
require('./index-b8d3e2ae.js');
require('./form-9a4efb54.js');
require('./dom-0693f998.js');
require('./guid-038573c1.js');
require('./resources-8bcff513.js');
require('./interactive-85bedd26.js');
require('./browser-49417561.js');
require('./key-dfee1d4b.js');
require('./label-9501231e.js');
require('./component-d4a67547.js');
require('./loadable-31c6cc1d.js');
require('./locale-8979e43c.js');
require('./observers-40b2a757.js');
require('./focusTrapComponent-e059d579.js');
require('./config-941721ae.js');
require('./math-05b8d5a7.js');
require('./t9n-166a2d65.js');
require('./Validation-e4bf1278.js');
require('./input-00c8a7d1.js');

// Japanese [ja]
var locale = {
  name: 'ja',
  weekdays: '日曜日_月曜日_火曜日_水曜日_木曜日_金曜日_土曜日'.split('_'),
  weekdaysShort: '日_月_火_水_木_金_土'.split('_'),
  weekdaysMin: '日_月_火_水_木_金_土'.split('_'),
  months: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  ordinal: function ordinal(n) {
    return n + "\u65E5";
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'YYYY/MM/DD',
    LL: 'YYYY年M月D日',
    LLL: 'YYYY年M月D日 HH:mm',
    LLLL: 'YYYY年M月D日 dddd HH:mm',
    l: 'YYYY/MM/DD',
    ll: 'YYYY年M月D日',
    lll: 'YYYY年M月D日 HH:mm',
    llll: 'YYYY年M月D日(ddd) HH:mm'
  },
  meridiem: function meridiem(hour) {
    return hour < 12 ? '午前' : '午後';
  },
  relativeTime: {
    future: '%s後',
    past: '%s前',
    s: '数秒',
    m: '1分',
    mm: '%d分',
    h: '1時間',
    hh: '%d時間',
    d: '1日',
    dd: '%d日',
    M: '1ヶ月',
    MM: '%dヶ月',
    y: '1年',
    yy: '%d年'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
