// @esri/calcite-components/dist/cjs/mk-253ce42c.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.12.1
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-6a00a289.js');
require('./index-b8d3e2ae.js');
require('./form-9a4efb54.js');
require('./dom-0693f998.js');
require('./guid-038573c1.js');
require('./resources-8bcff513.js');
require('./interactive-85bedd26.js');
require('./browser-49417561.js');
require('./key-dfee1d4b.js');
require('./label-9501231e.js');
require('./component-d4a67547.js');
require('./loadable-31c6cc1d.js');
require('./locale-8979e43c.js');
require('./observers-40b2a757.js');
require('./focusTrapComponent-e059d579.js');
require('./config-941721ae.js');
require('./math-05b8d5a7.js');
require('./t9n-166a2d65.js');
require('./Validation-e4bf1278.js');
require('./input-00c8a7d1.js');

// Macedonian [mk]
var locale = {
  name: 'mk',
  weekdays: 'недела_понеделник_вторник_среда_четврток_петок_сабота'.split('_'),
  months: 'јануари_февруари_март_април_мај_јуни_јули_август_септември_октомври_ноември_декември'.split('_'),
  weekStart: 1,
  weekdaysShort: 'нед_пон_вто_сре_чет_пет_саб'.split('_'),
  monthsShort: 'јан_фев_мар_апр_мај_јун_јул_авг_сеп_окт_ное_дек'.split('_'),
  weekdaysMin: 'нe_пo_вт_ср_че_пе_сa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY H:mm',
    LLLL: 'dddd, D MMMM YYYY H:mm'
  },
  relativeTime: {
    future: 'после %s',
    past: 'пред %s',
    s: 'неколку секунди',
    m: 'минута',
    mm: '%d минути',
    h: 'час',
    hh: '%d часа',
    d: 'ден',
    dd: '%d дена',
    M: 'месец',
    MM: '%d месеци',
    y: 'година',
    yy: '%d години'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
