// @esri/calcite-components/dist/esm/es-454645a3.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.12.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-2a0e039e.js';
import './index-590f87c2.js';
import './form-0d9970ac.js';
import './dom-0dfcce02.js';
import './guid-c5ea645c.js';
import './resources-988e2a52.js';
import './interactive-7dc921d0.js';
import './browser-b3aa6d44.js';
import './key-6b945008.js';
import './label-3e643a13.js';
import './component-3035aa13.js';
import './loadable-907717a5.js';
import './locale-4ce3c6b2.js';
import './observers-aea7721f.js';
import './focusTrapComponent-4c78325e.js';
import './config-df1b58be.js';
import './math-db618c8a.js';
import './t9n-249fa143.js';
import './Validation-c0354df6.js';
import './input-30caec30.js';

// Spanish [es]
var locale = {
  name: 'es',
  monthsShort: 'ene_feb_mar_abr_may_jun_jul_ago_sep_oct_nov_dic'.split('_'),
  weekdays: 'domingo_lunes_martes_miércoles_jueves_viernes_sábado'.split('_'),
  weekdaysShort: 'dom._lun._mar._mié._jue._vie._sáb.'.split('_'),
  weekdaysMin: 'do_lu_ma_mi_ju_vi_sá'.split('_'),
  months: 'enero_febrero_marzo_abril_mayo_junio_julio_agosto_septiembre_octubre_noviembre_diciembre'.split('_'),
  weekStart: 1,
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD/MM/YYYY',
    LL: 'D [de] MMMM [de] YYYY',
    LLL: 'D [de] MMMM [de] YYYY H:mm',
    LLLL: 'dddd, D [de] MMMM [de] YYYY H:mm'
  },
  relativeTime: {
    future: 'en %s',
    past: 'hace %s',
    s: 'unos segundos',
    m: 'un minuto',
    mm: '%d minutos',
    h: 'una hora',
    hh: '%d horas',
    d: 'un día',
    dd: '%d días',
    M: 'un mes',
    MM: '%d meses',
    y: 'un año',
    yy: '%d años'
  },
  ordinal: function ordinal(n) {
    return n + "\xBA";
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
