// @esri/calcite-components/dist/esm/it-ch-e6cfac79.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.12.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-2a0e039e.js';
import './index-590f87c2.js';
import './form-0d9970ac.js';
import './dom-0dfcce02.js';
import './guid-c5ea645c.js';
import './resources-988e2a52.js';
import './interactive-7dc921d0.js';
import './browser-b3aa6d44.js';
import './key-6b945008.js';
import './label-3e643a13.js';
import './component-3035aa13.js';
import './loadable-907717a5.js';
import './locale-4ce3c6b2.js';
import './observers-aea7721f.js';
import './focusTrapComponent-4c78325e.js';
import './config-df1b58be.js';
import './math-db618c8a.js';
import './t9n-249fa143.js';
import './Validation-c0354df6.js';
import './input-30caec30.js';

// Italian (Switzerland) [it-ch]
var locale = {
  name: 'it-ch',
  weekdays: 'domenica_lunedì_martedì_mercoledì_giovedì_venerdì_sabato'.split('_'),
  months: 'gennaio_febbraio_marzo_aprile_maggio_giugno_luglio_agosto_settembre_ottobre_novembre_dicembre'.split('_'),
  weekStart: 1,
  weekdaysShort: 'dom_lun_mar_mer_gio_ven_sab'.split('_'),
  monthsShort: 'gen_feb_mar_apr_mag_giu_lug_ago_set_ott_nov_dic'.split('_'),
  weekdaysMin: 'do_lu_ma_me_gi_ve_sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'tra %s',
    past: '%s fa',
    s: 'alcuni secondi',
    m: 'un minuto',
    mm: '%d minuti',
    h: 'un\'ora',
    hh: '%d ore',
    d: 'un giorno',
    dd: '%d giorni',
    M: 'un mese',
    MM: '%d mesi',
    y: 'un anno',
    yy: '%d anni'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
