// @esri/calcite-components/dist/esm/ja-6a432ef4.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.12.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-2a0e039e.js';
import './index-590f87c2.js';
import './form-0d9970ac.js';
import './dom-0dfcce02.js';
import './guid-c5ea645c.js';
import './resources-988e2a52.js';
import './interactive-7dc921d0.js';
import './browser-b3aa6d44.js';
import './key-6b945008.js';
import './label-3e643a13.js';
import './component-3035aa13.js';
import './loadable-907717a5.js';
import './locale-4ce3c6b2.js';
import './observers-aea7721f.js';
import './focusTrapComponent-4c78325e.js';
import './config-df1b58be.js';
import './math-db618c8a.js';
import './t9n-249fa143.js';
import './Validation-c0354df6.js';
import './input-30caec30.js';

// Japanese [ja]
var locale = {
  name: 'ja',
  weekdays: '日曜日_月曜日_火曜日_水曜日_木曜日_金曜日_土曜日'.split('_'),
  weekdaysShort: '日_月_火_水_木_金_土'.split('_'),
  weekdaysMin: '日_月_火_水_木_金_土'.split('_'),
  months: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  ordinal: function ordinal(n) {
    return n + "\u65E5";
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'YYYY/MM/DD',
    LL: 'YYYY年M月D日',
    LLL: 'YYYY年M月D日 HH:mm',
    LLLL: 'YYYY年M月D日 dddd HH:mm',
    l: 'YYYY/MM/DD',
    ll: 'YYYY年M月D日',
    lll: 'YYYY年M月D日 HH:mm',
    llll: 'YYYY年M月D日(ddd) HH:mm'
  },
  meridiem: function meridiem(hour) {
    return hour < 12 ? '午前' : '午後';
  },
  relativeTime: {
    future: '%s後',
    past: '%s前',
    s: '数秒',
    m: '1分',
    mm: '%d分',
    h: '1時間',
    hh: '%d時間',
    d: '1日',
    dd: '%d日',
    M: '1ヶ月',
    MM: '%dヶ月',
    y: '1年',
    yy: '%d年'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
